---
title: "Re-run EWAS Log 1_08Apr23"
author: "Jasmine Douglas"
date: "2023-04-08"
output: html_document
---
## import the data

Load the libraries and change the memory limit.
```{r}
library(ewastools)
library(stringi)
library(data.table)
library(magrittr)
library(purrr)
library(limma)
library(svd)
library(ggplot2)
memory.limit(size=50000)
```

1. Read in the file pheno.csv using fread function from the data.table package.
2. Import the methylation data using the function read_idats with the filename before _Red.idat.gz or _Grn.idat.gz, which saved in pheno$dataid column. 
```{r, warning=FALSE}
pheno = fread("C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\MICS Epigenetics Data\\MICS\\MICS\\pheno.csv")

meth = read_idats("C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\MICS Epigenetics Data\\MICS\\MICS\\All_raw_data\\" %s+% pheno$dataid,quiet=TRUE)
```

check the basic information
1. Check the platform first. Our data platform is EPIC.
2. Take a look on the manifest to see the probe ID, color channel, genomic coordinates.
```{r}
meth$platform
meth$manifest[4000:4010]
```
3. Take a look on the channel
4. Type I probes have addressU and addressM, type II probes only have a single addressU. 
5. Not all probes are targeting CpG sites
```{r}
table(meth$manifest$channel)
table(meth$manifest$probe_type)
```

6. Take a look on the manifest for control probes
```{r}
head(meth$controls)
```

Matrices contained fluorescence intensities for the methylated (`M`) and unmethylated (`U`) signals
```{r}
dim(meth$M)
meth$U[201:203,1:3]
meth$M[201:203,1:3]
```

Matrices with the bead copy number (N and V)
```{r}
dim(meth$N)
meth$N[201:203,1:3]
```

Check the beads that the copy number is zero. It is because of the random assembly some did not end up on the chips
```{r}
colSums(meth$N==0)
```

Matrices with the out-of-band intensities
```{r}
dim(meth$oobG$M)
dim(meth$oobR$U)
```

Some meta data
```{r}
head(meth$meta)
```

## Quality control**

# Control metrics*

To test the 17 control metrics whether fail to pass the quality control described in the BeadArray Controls Reporter Software Guide from Illumina.
The result is logical, and return 'TRUE' if sample at corresponding index failed on any of the 17 control metrics.

```{r}
meth %>% control_metrics %>% sample_failure -> pheno$failed
table(pheno$failed,useNA='ifany')
```

# Sex check

Check the predicted sex with the true sex in the pheno table.\n
1. generate a column called predicted_sex to save the prediction of sex results.
2. create a new column called sex_letter to save the true sex information in "m" or "f" format.
3. compare predicted_sex vs. sex_letter to find the differences.

P.S.: there are 5 samples don't have a gender information.\n
result: No sex information are different from prediction results. 
The five unknown sex people all predicted as Female.\n
No sample will be exclude out.

```{r}
pheno[,exclude:=FALSE]
pheno[,c("X","Y"):=check_sex(meth)]
pheno$predicted_sex = predict_sex(pheno$X,pheno$Y,which(pheno$Sex=="Male"),which(pheno$Sex=="Female"))
pheno$Sex_letter <- gsub('Male', 'm', pheno$Sex)
pheno$Sex_letter <- gsub('Female', 'f', pheno$Sex_letter)
pheno[Sex_letter!=predicted_sex][,1:3]
```

In the plot, the X and Y-axis is the predicted sex, and the text in the plot is the true sex. There is one sample true sex is Female but predicted as Male.

```{r}
plot(Y~X,data=pheno,type="n")
text(Y~X,labels=Sex_letter,col=ifelse(Sex_letter=="m",2,1),data=pheno)
```


# Detection p-values

```{r}
meth = ewastools::detectionP(meth)
chrY = meth$manifest[chr=='Y',index]
detP = meth$detP[chrY,]
detP = colSums(detP<0.01,na.rm=TRUE)

boxplot(split(detP,pheno$predicted_sex),ylab="# of detected Y chromosome probes")
```

```{r}
split(detP,pheno$predicted_sex) %>% sapply(mean)
```

Almost all of the 534 chromosome probes are called detected in male samples, for female samples on average 110 are called detected. How many probes are undetected (not counting the Y chromosome). The cut-off used here is 0.01

```{r}
meth$detP[-chrY,] %>% is_weakly_greater_than(0.01) %>% table(useNA="ifany")
```

About 0.25% are undetected \n 

```{r}
round((407615/(407615+163958345))*100,3)
```

We mask these undetected probes.
```{r}
meth = ewastools::mask(meth,0.01)
```



**Dye-bias correction**

The A and T nucleotides linked with red dye, G and C linked with green dye. The red color channel usually has higher intensities without corrected data. For probes of Infinium type II design, which use separate color channels to measure the methylated and unmethylated signal, this results in a shifted distribution of beta-values. (Probes of Infinium design type I are not affected, as they measure both signals in the same color channel.) 

We used ewastools to do the dye-bias correction.
```{r}
color_bias = meth %>% dont_normalize
beta       = meth %>% correct_dye_bias %>% dont_normalize
```

Take a look on the data to see whether dye-bias correction worked.
```{r}
meth$manifest$channel[2:4]
color_bias[2:4,1:3] %>% round(4)
beta      [2:4,1:3] %>% round(4)
```

If we calculate beta-values from raw data, we can observe the dye bias as a deviation of the beta-values for heterozygous SNPs from 0.47

```{r}
snps = meth$manifest[probe_type=="rs" & channel=="Both"]$index

plot (density(color_bias[snps,14],na.rm=TRUE,bw=0.1),col=1,main="Dye-bias correction")
lines(density(beta      [snps,14],na.rm=TRUE,bw=0.1),col=2)
abline(v=0.47,lty=3)
legend("topleft",col=1:2,legend=c("raw","corrected"),lwd=1)

plot (density(beta[meth$manifest$channel=="Grn" ,1],na.rm=TRUE),col="green",main="Distribution of beta-values")
lines(density(beta[meth$manifest$channel=="Red" ,1],na.rm=TRUE),col="red")
lines(density(beta[meth$manifest$channel=="Both",1],na.rm=TRUE),col="black")
legend("topright",legend=c("Type I Red","Type I Grn","Type II"),lwd=1,col=c("red","green","black"))
```

# SNP outliers
Check the abnormal average log odds across all SNP probes.
I used -3.5 as threshold. Exclude it if it's higher than -3.5. 
```{r}
snps = meth$manifest[probe_type=="rs"]$index
genotypes = call_genotypes(beta[snps,],learn=FALSE)
pheno$outlier = snp_outliers(genotypes)

stripchart(pheno$outlier,method="jitter",pch=4)
abline(v=-3.5,lty="dotted",col=2)
```

If I use -4 as threshold like the author suggested, there are two samples excluded. \n
However, we decided to keep one, and drop the other, so -3.5 was used here. \n

```{r}
pheno[outlier>-3.5]
pheno[outlier>-3.5,exclude:=TRUE]

pheno$donor_id = enumerate_sample_donors(genotypes)
pheno[,n:=.N,by=donor_id]
pheno[n>1,.('External Reference',donor_id)]
```

# Principal component analysis

Principal component analysis is a popular feature reduction method: it projects high-dimensional data into a lower-dimensional representation while trying to retain as much variability as possible. This is especially useful when either individual features are highly correlated and it is therefore reasonable to summarize them, or when (sometimes subtle) traces of background effects can be found across of large number of features. 1. Get of subset of beta without probes on the X and Y chromosome

1. Get of subset of beta without probes on the X and Y chromosome.\n
2. Plot them on PCA plot and label the excluded samples.

male = black female = red

The NO. 188 is from the outlier result. 

```{r}
chrXY = meth$manifest$chr %in% c("X","Y") | meth$manifest$probe_type == "rs"
pcs = beta[-chrXY,]
pcs = pcs - rowMeans(pcs)
pcs = na.omit(pcs)
pcs = t(pcs)
pcs = trlan.svd(pcs,neig=2)
pcs = pcs$u

pheno$pc1 = pcs[,1]
pheno$pc2 = pcs[,2]

plot(pc2~pc1,col=ifelse(predicted_sex=="m",1,2),data=pheno)

text(pc2~pc1,labels=pheno[188]$SampleID,data=pheno[188],pos=4,offset=1,col=2)

```


# Leukocyte composition
This quality check only apply in case of blood samples.Use estimateLC function to predict the leukocyte composition through Houseman method.

```{r}
LC = estimateLC(beta,ref="Reinius")
stripchart(rowSums(LC),xlab="Sum of cell proportions",m="jitter")
abline(v=0.95,lty=3,col=2)

pheno = cbind(pheno,LC)
plot(pheno$CD4,ylim=c(0,1))
plot(pheno$CD8,ylim=c(0,1))
plot(pheno$NK,ylim=c(0,1))
plot(pheno$B,ylim=c(0,1))
plot(pheno$MO,ylim=c(0,1))
# pheno[which.min(CD4),.(dataid,exclude)]
# pheno[which.max(CD4),.(dataid,exclude)]
```

LC stratified by Group
```{r}
LC$Group = as.factor(pheno$Group)
LC = melt(LC,value.name="proportion",variable.name="cell_type",id.vars="Group")

boxplot(proportion ~ Group+cell_type,LC,col=1:3,main="Cell type distribution by status",xaxt="n")
axis(1,at=seq(from=2, to=17,by=3),adj=1,labels=unique(LC$cell_type))
legend("topright",levels(LC$Group), pch=15,bty='n',col=1:3)

 
ggplot(LC, aes(x=cell_type, y=proportion, fill=Group)) + 
    geom_boxplot()
 
```

Drop 2 excluded data
```{r}
keep = which(!pheno$exclude)
```

Keep the columns we need
1. 189 samples were keep.
2. Keep the important columns in pheno
3. 865918 sites were obtained, included 862927 cg sites, 2932 ch sites and 59 rs sites.
```{r}
pheno = pheno[,.(SampleID,External_Reference,SubjectID,Group,UniqueNum,Month,Age_in_m,Age,Type,predicted_sex,Sex,Race,Ethnicity,Num,Timepoints,dataid,CD4,CD8,NK,MO,GR,B)]

pheno = pheno[keep]
dim(pheno)
manifest = copy(meth$manifest)
dim(manifest)
beta  = beta[,keep]
dim(beta)
```

Save the new pheno table as pheno_clean.csv, and save the pheno, manifest and beta together as processed.rda for further use.
```{r}
save(pheno,manifest,beta,file="C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\Jasmine\\Aim 1\\processed_one_outlier_dropped_rerun08Apr23.rda")
write.csv(pheno, file = "C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\Jasmine\\Aim 1\\pheno-clean_one_outlier_dropped_rerun08Apr23.csv")
```
