---
title: "EWAS-log2_MICS_one-outlier-dropped_targeted_adjusted sex, race, cell type"
author: "Jasmine Douglas"
date: "2024-02-01"
output: html_document
---


#Load required libraries and the rda file produced in pre-processing (log1) that includes beta, pheno, and manifest information.

```{r}
suppressPackageStartupMessages({
  library(DMRcate) # for regional analysis
  library(magrittr)
  library(CpGassoc)#for running association analysis between methylation levels values and phenotype 
  library(data.table) # for fast aggregation of large data 
  library(qqman) # for visualization of data
  library(stringi) # string manipulation
  library(dplyr)
  library(ggplot2)
  library(minfi)
  library(limma)
})
```

```{r}
load("C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\Jasmine\\Aim 1\\processed_one_outlier_dropped_rerun08Apr23.rda")
dim(beta)
```

Filter a matrix of betas values by distance to single nucleotide polymorphisms (SNPs) and by SNPs with minor allele frequency (MAF) of 5% (rare variant). Also removes crosshybridising and sex chromosome probes; named beta.clean. After removing the sex chromosome probes, there were 843,393 CpG sites left.

```{r}
betas.clean = beta[manifest[probe_type == "cg" & !chr %in% c("X","Y")]$index,]
dim(betas.clean)
rm(beta)
nCpG = dim(betas.clean)[1]
nCpG
```


Add variables that will be adjusted for in the analysis. Need to add race_eth, predicted_sex to existing pheno table.

```{r}
pheno.a <-read.csv("C://Users//jasmi//Box//RU MICS (Shared)//MICS Epigenetics Data//MICS//MICS//pheno.csv")
pheno.a <-pheno.a[,c("External_Reference","race_eth")]
pheno.a$race_eth.new <- ifelse((pheno.a$race_eth == 'Non-Hispanic Black'), 'Non-Hispanic Black', "Other")
pheno<-left_join(pheno, pheno.a, by="External_Reference")
dim(pheno)
```


Code categorical variables for analysis as a factor. Required for CpGassoc.

```{r}
library(forcats)
pheno$Timepoints %<>% factor

pheno$race_eth<-as_factor(pheno$race_eth)
pheno$predicted_sex<-as_factor(pheno$predicted_sex)
class(pheno$race_eth)
class(pheno$predicted_sex)
```


## Code - Timepoint 3mo. - HIV vs. HEU, Sex & Race & ct5 (CD8, CD4, MO, NK, GR)

#Group - hiv vs. heu - Adjusted for race, sex, cell type 

Subset Timepoint 3 and compare group hiv & heu for both pheno and betas.clean. 57 samples were extract5ed.

hiv & heu at Timepoint 3 = 24 vs. 33.

```{r}
tp3_hiv_heu <- which(pheno$Timepoints==3 & pheno$Group!="HUU")
pheno.tp3_hiv_heu <- pheno[tp3_hiv_heu]
betas.clean.tp3_hiv_heu = betas.clean[,tp3_hiv_heu]
table(pheno.tp3_hiv_heu$Group)

```


Remove cpg sites from betas.clean that cause cpg.assoc errors.

```{r}
library(stringr)

#retrieving dataids for male and female samples.
pheno.tp3_hiv_heu[which(predicted_sex=='m'),] %>% select(dataid) -> m1
pheno.tp3_hiv_heu[which(predicted_sex=='f'),] %>% select(dataid) -> f1

##adding file/data location to align with clean.betas columns.
m1$file.loc = "C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\MICS Epigenetics Data\\MICS\\MICS\\All_raw_data"
f1$file.loc = "C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\MICS Epigenetics Data\\MICS\\MICS\\All_raw_data"

str_c(m1$file.loc, m1$dataid, sep = "\\") -> m1$betas.id ->m2
str_c(f1$file.loc, f1$dataid, sep = "\\") -> f1$betas.id ->f2

##creating vectors of male and female samples.
print(m2)
as.vector(m2)->m2
as.vector(f2)->f2

#Subset betas.clean for male and female samples. Remove sites for CpGs with all NAs for males or females.
betas.clean.new.m <- betas.clean.tp3_hiv_heu[,m2]
betas.clean.new.f <- betas.clean.tp3_hiv_heu[,f2]

remove.m <- which(rowSums(is.na(betas.clean.new.m)) == length(m1$dataid))
remove.f <- which(rowSums(is.na(betas.clean.new.f)) == length(f1$dataid))

betas.new.tp3_hiv_heu <- betas.clean.tp3_hiv_heu[-c(remove.m,remove.f),]

#total number of sites removed.
nrow(betas.clean.tp3_hiv_heu) - nrow(betas.new.tp3_hiv_heu)

```


Analyze the association between methylation beta values using betas.new and the phenotype of interest. Results show association between methylation beta values and Group column.

Set up a dummy variable that HIV+=1, HEU=0

```{r}
pheno.tp3_hiv_heu$hiv.cat= ifelse(pheno.tp3_hiv_heu$Group == "HIV+",1,0)

source("C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\Ziyi\\Code\\cpg.assoc_mod.R")

resultstp3_hiv_heu.sex.race.ct5 = cpg.assoc_mod(betas.new.tp3_hiv_heu,
                                                 pheno.tp3_hiv_heu$hiv.cat,
                                                 covariates=as.data.frame(pheno.tp3_hiv_heu[,.(predicted_sex,race_eth.new,CD8,CD4,NK,GR,MO)]
                                                 )
)
class(resultstp3_hiv_heu.sex.race.ct5)
names(resultstp3_hiv_heu.sex.race.ct5)
print(resultstp3_hiv_heu.sex.race.ct5)
```

Check the top hits by ordering the p.value.
The effect size here is ~ mean difference in methylation proportion.
```{r}
head(cbind(resultstp3_hiv_heu.sex.race.ct5$coefficients, P.value=resultstp3_hiv_heu.sex.race.ct5$results[,3])[order(resultstp3_hiv_heu.sex.race.ct5$results[,3]),])
```

Check Bonferroni significant sites of resultstp3_hiv_heu.sex.race.ct5 using p-value/ncpg.

```{r}
table(resultstp3_hiv_heu.sex.race.ct5$results[,3]< 0.05/nCpG)
```
#Genomic Inflation

Combine the sites with the manifest and view top sites. Use the combined data to create Volcano plot.

```{r}
datamanhattp3_hiv_heu.sex.race.ct5 = cbind(resultstp3_hiv_heu.sex.race.ct5$results, resultstp3_hiv_heu.sex.race.ct5$coefficients)
setDT(datamanhattp3_hiv_heu.sex.race.ct5)
datamanhattp3_hiv_heu.sex.race.ct5 = datamanhattp3_hiv_heu.sex.race.ct5[,.(probe_id=CPG.Labels,effect.size,std.error,P.value,FDR)]

datamanhattp3_hiv_heu.sex.race.ct5 = merge(datamanhattp3_hiv_heu.sex.race.ct5,manifest[,.(probe_id,chr,mapinfo)],by="probe_id")
datamanhattp3_hiv_heu.sex.race.ct5[order(P.value)][1:10]
```

Volcano plot of resultstp3_hiv_heu using Bonferroni threshold. 

2 sites met the Bonferroni threshold. 

#Volcano Plot
```{r}
with(datamanhattp3_hiv_heu.sex.race.ct5, plot(effect.size, -log10(P.value), pch=20, main=""))
abline(h = -log10(0.05/(nCpG)), col = "blue", lty = 2, lwd = 1)
# abline(h = -log10(2.5e-07), col = "blue", lty = 2, lwd = 1)
abline(v = c(-0.05,0.05), col = "black", lty = 2, lwd = 1)

with(subset(datamanhattp3_hiv_heu.sex.race.ct5, effect.size< -0.05 & FDR < 0.05), points(effect.size, -log10(P.value), pch=20, col="red"))

with(subset(datamanhattp3_hiv_heu.sex.race.ct5, effect.size > 0.05 & FDR < 0.05), points(effect.size, -log10(P.value), pch=20, col="green"))
```

Cast the variable "chr" or chromosome to view significant sites by chromosome.

#Manhattan Plot
```{r}
datamanhattp3_hiv_heu.sex.race.ct5[,chr:=as.integer(chr)]

qqman::manhattan(datamanhattp3_hiv_heu.sex.race.ct5,chr="chr",bp="mapinfo",p="P.value",snp="probe_id"
                 ,suggestiveline=FALSE, genomewideline = -log10(0.05/(nCpG)),ylim=c(0,15)
                 ,main = "Manhattan Plot of mean DNA methylation diference\nbetween HIV+ and HEU at Timepoint 3 - Adjusted")
```

#Limma

Use lmfit to fit a linear model to clean betas vs. heu status. Use fitted linear model to compute moderated t & F statistics, and log odds by empirical Bayes moderation of standard errors toward a global value.

```{r}
library(limma,minfi)

pheno.tp3_hiv_heu$Group <- factor(pheno.tp3_hiv_heu$Group, levels = c("HIV+", "HEU"))
model = model.matrix( ~Group+predicted_sex+race_eth.new+CD8+CD4+NK+GR+MO,data=pheno.tp3_hiv_heu)
EWAS.limma <- eBayes(lmFit(betas.new.tp3_hiv_heu, design=model))
```

Check the top sites with statistically significant differential methylation across groups.

```{r}
Toptp3_hiv_heu.sex.race.ct5<-topTable(EWAS.limma, coef=2, number=Inf, sort.by = "p")

head(Toptp3_hiv_heu.sex.race.ct5)
```

Calculate the values of the two groups using getEWAP function. Generate dataframe called "average" to save top 500 sites for differential methylation between the two groups.

```{r}
genenames<-rownames(Toptp3_hiv_heu.sex.race.ct5)
ewap<-getEAWP(betas.new.tp3_hiv_heu)
average<- data.frame(hiv=rep(0,500), heu=rep(0,500))
for (i in 1:1500){average[i,]<-tapply(ewap$exprs[genenames[i],], pheno.tp3_hiv_heu$Group, mean, na.rm=TRUE)
}
average$CPG.Labels <- genenames[1:1500]
```

1. Use IlluminaHumanMethylationEPICanno.ilm10b4.hg19 to annotate the top sites with thier associated genes. 
2. Match TopTable with Annot output.
3. Remove unnecessary information.

```{r}
require(IlluminaHumanMethylationEPICanno.ilm10b4.hg19)

Annot<-as.data.frame(getAnnotation(IlluminaHumanMethylationEPICanno.ilm10b4.hg19))
Annot.Tops<- Annot[match(rownames(Toptp3_hiv_heu.sex.race.ct5),Annot$Name),]
Annot.Tops<-Annot.Tops[,c("UCSC_RefGene_Name","UCSC_RefGene_Group","Relation_to_Island","chr","pos")]
Toptp3_hiv_heu.sex.race.ct5<-cbind(Toptp3_hiv_heu.sex.race.ct5, Annot.Tops)
head(Toptp3_hiv_heu.sex.race.ct5)

```

```{r}
Toptp3_hiv_heu.sex.race.ct5 <- tibble::rownames_to_column(Toptp3_hiv_heu.sex.race.ct5, "CPG.Labels")
Toptp3_hiv_heu.sex.race.ct5 <- left_join(Toptp3_hiv_heu.sex.race.ct5, resultstp3_hiv_heu.sex.race.ct5$results, by = "CPG.Labels")

toptp3_hiv_heu.sex.race.ct5 <- left_join(Toptp3_hiv_heu.sex.race.ct5, average, by = "CPG.Labels")

```



## Code - Timepoint 3mo. - HIV vs. huu, Sex & Race & ct5 (CD8, CD4, MO, NK, GR)

#Group - hiv vs. huu - Adjusted for race, sex, cell type 

Subset Timepoint 3 and compare group hiv & huu for both pheno and betas.clean. 57 samples were extracted

hiv & huu at Timepoint 3 = 24 vs. 33.

```{r}
tp3_hiv_huu <- which(pheno$Timepoints==3 & pheno$Group!="HEU")
pheno.tp3_hiv_huu <- pheno[tp3_hiv_huu]
betas.clean.tp3_hiv_huu = betas.clean[,tp3_hiv_huu]
table(pheno.tp3_hiv_huu$Group)

```



Remove cpg sites from betas.clean that cause cpg.assoc errors.

```{r}
library(stringr)

#retrieving dataids for male and female samples.
pheno.tp3_hiv_huu[which(predicted_sex=='m'),] %>% select(dataid) -> m3
pheno.tp3_hiv_huu[which(predicted_sex=='f'),] %>% select(dataid) -> f3

##adding file/data location to align with clean.betas columns.
m3$file.loc = "C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\MICS Epigenetics Data\\MICS\\MICS\\All_raw_data"
f3$file.loc = "C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\MICS Epigenetics Data\\MICS\\MICS\\All_raw_data"

str_c(m3$file.loc, m3$dataid, sep = "\\") -> m3$betas.id ->m4
str_c(f3$file.loc, f3$dataid, sep = "\\") -> f3$betas.id ->f4

##creating vectors of male and female samples.
print(m4)
as.vector(m4)->m4
as.vector(f4)->f4

#Subset betas.clean for male and female samples. Remove sites for CpGs with all NAs for males or females.
betas.clean.new.m1 <- betas.clean.tp3_hiv_huu[,m4]
betas.clean.new.f1 <- betas.clean.tp3_hiv_huu[,f4]

remove.m1 <- which(rowSums(is.na(betas.clean.new.m1)) == length(m3$dataid))
remove.f1 <- which(rowSums(is.na(betas.clean.new.f1)) == length(f3$dataid))

betas.new.tp3_hiv_huu <- betas.clean.tp3_hiv_huu[-c(remove.m1,remove.f1),]

#total number of sites removed.
nrow(betas.clean.tp3_hiv_huu) - nrow(betas.new.tp3_hiv_huu)

```

Analyze the association between methylation beta values using betas.new and the phenotype of interest. Results show association between methylation beta values and Group column.

Set up a dummy variable that HIV+=1, HUU=0

```{r}
pheno.tp3_hiv_huu$hiv.cat= ifelse(pheno.tp3_hiv_huu$Group == "HIV+",1,0)

source("C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\Ziyi\\Code\\cpg.assoc_mod.R")

resultstp3_hiv_huu.sex.race.ct5 = cpg.assoc_mod(betas.new.tp3_hiv_huu,
                                                 pheno.tp3_hiv_huu$hiv.cat,
                                                 covariates=as.data.frame(pheno.tp3_hiv_huu[,.(predicted_sex,race_eth.new,CD8,CD4,NK,GR,MO)]
                                                 )
)
class(resultstp3_hiv_huu.sex.race.ct5)
names(resultstp3_hiv_huu.sex.race.ct5)
print(resultstp3_hiv_huu.sex.race.ct5)
```

Check the top hits by ordering the p.value.
The effect size here is ~ mean difference in methylation proportion.
```{r}
head(cbind(resultstp3_hiv_huu.sex.race.ct5$coefficients, P.value=resultstp3_hiv_huu.sex.race.ct5$results[,3])[order(resultstp3_hiv_huu.sex.race.ct5$results[,3]),])
```

Check Bonferroni significant sites of resultstp3_hiv_huu.sex.race.ct5 using p-value/ncpg.

```{r}
table(resultstp3_hiv_huu.sex.race.ct5$results[,3]< 0.05/nCpG)
```
#Genomic Inflation

Combine the sites with the manifest and view top sites. Use the combined data to create Volcano plot.

```{r}
datamanhattp3_hiv_huu.sex.race.ct5 = cbind(resultstp3_hiv_huu.sex.race.ct5$results, resultstp3_hiv_huu.sex.race.ct5$coefficients)
setDT(datamanhattp3_hiv_huu.sex.race.ct5)
datamanhattp3_hiv_huu.sex.race.ct5 = datamanhattp3_hiv_huu.sex.race.ct5[,.(probe_id=CPG.Labels,effect.size,std.error,P.value,FDR)]

datamanhattp3_hiv_huu.sex.race.ct5 = merge(datamanhattp3_hiv_huu.sex.race.ct5,manifest[,.(probe_id,chr,mapinfo)],by="probe_id")
datamanhattp3_hiv_huu.sex.race.ct5[order(P.value)][1:10]
```

Volcano plot of resultstp3_hiv_huu using Bonferroni threshold. 

2 sites met the Bonferroni threshold. 

#Volcano Plot
```{r}
with(datamanhattp3_hiv_huu.sex.race.ct5, plot(effect.size, -log10(P.value), pch=20, main=""))
abline(h = -log10(0.05/(nCpG)), col = "blue", lty = 2, lwd = 1)
# abline(h = -log10(2.5e-07), col = "blue", lty = 2, lwd = 1)
abline(v = c(-0.05,0.05), col = "black", lty = 2, lwd = 1)

with(subset(datamanhattp3_hiv_huu.sex.race.ct5, effect.size< -0.05 & FDR < 0.05), points(effect.size, -log10(P.value), pch=20, col="red"))

with(subset(datamanhattp3_hiv_huu.sex.race.ct5, effect.size > 0.05 & FDR < 0.05), points(effect.size, -log10(P.value), pch=20, col="green"))
```

Cast the variable "chr" or chromosome to view significant sites by chromosome.

#Manhattan Plot
```{r}
datamanhattp3_hiv_huu.sex.race.ct5[,chr:=as.integer(chr)]

qqman::manhattan(datamanhattp3_hiv_huu.sex.race.ct5,chr="chr",bp="mapinfo",p="P.value",snp="probe_id"
                 ,suggestiveline=FALSE, genomewideline = -log10(0.05/(nCpG)),ylim=c(0,15)
                 ,main = "Manhattan Plot of mean DNA methylation diference\nbetween HIV+ and huu at Timepoint 3 - Adjusted")
```

#Limma

Use lmfit to fit a linear model to clean betas vs. huu status. Use fitted linear model to compute moderated t & F statistics, and log odds by empirical Bayes moderation of standard errors toward a global value.

```{r}
library(limma,minfi)

pheno.tp3_hiv_huu$Group <- factor(pheno.tp3_hiv_huu$Group, levels = c("HIV+", "HUU"))
model = model.matrix( ~Group+predicted_sex+race_eth.new+CD8+CD4+NK+GR+MO,data=pheno.tp3_hiv_huu)
EWAS.limma <- eBayes(lmFit(betas.new.tp3_hiv_huu, design=model))
```

Check the top sites with statistically significant differential methylation across groups.

```{r}
Toptp3_hiv_huu.sex.race.ct5<-topTable(EWAS.limma, coef=2, number=Inf, sort.by = "p")

head(Toptp3_hiv_huu.sex.race.ct5)
```

Calculate the values of the two groups using getEWAP function. Generate dataframe called "average" to save top 500 sites for differential methylation between the two groups.

```{r}
genenames<-rownames(Toptp3_hiv_huu.sex.race.ct5)
ewap<-getEAWP(betas.new.tp3_hiv_huu)
average<- data.frame(hiv=rep(0,500), huu=rep(0,500))
for (i in 1:1500){average[i,]<-tapply(ewap$exprs[genenames[i],], pheno.tp3_hiv_huu$Group, mean, na.rm=TRUE)
}
average$CPG.Labels <- genenames[1:1500]
```

1. Use IlluminaHumanMethylationEPICanno.ilm10b4.hg19 to annotate the top sites with thier associated genes. 
2. Match TopTable with Annot output.
3. Remove unnecessary information.

```{r}
require(IlluminaHumanMethylationEPICanno.ilm10b4.hg19)

Annot<-as.data.frame(getAnnotation(IlluminaHumanMethylationEPICanno.ilm10b4.hg19))
Annot.Tops<- Annot[match(rownames(Toptp3_hiv_huu.sex.race.ct5),Annot$Name),]
Annot.Tops<-Annot.Tops[,c("UCSC_RefGene_Name","UCSC_RefGene_Group","Relation_to_Island","chr","pos")]
Toptp3_hiv_huu.sex.race.ct5<-cbind(Toptp3_hiv_huu.sex.race.ct5, Annot.Tops)
head(Toptp3_hiv_huu.sex.race.ct5)

```

Save results. Save logFC >= .05 sites as well as FDR, Bonferroni significant sites.

```{r}
Toptp3_hiv_huu.sex.race.ct5 <- tibble::rownames_to_column(Toptp3_hiv_huu.sex.race.ct5, "CPG.Labels")
Toptp3_hiv_huu.sex.race.ct5 <- left_join(Toptp3_hiv_huu.sex.race.ct5, resultstp3_hiv_huu.sex.race.ct5$results, by = "CPG.Labels")
toptp3_hiv_huu.sex.race.ct5 <- left_join(Toptp3_hiv_huu.sex.race.ct5, average, by = "CPG.Labels")

```


## Code - Timepoint 3mo. - Heu vs. huu, Sex & Race & ct5 (CD8, CD4, MO, NK, GR)

#Group - Heu vs. Huu - Adjusted for race, sex, cell type 

Subset Timepoint 3 and compare group heu & huu for both pheno and betas.clean. 66 samples were extracted

heu & huu at Timepoint 3 = 33 vs. 33.

```{r}
tp3_heu_huu <- which(pheno$Timepoints==3 & pheno$Group!="HIV+")
pheno.tp3_heu_huu <- pheno[tp3_heu_huu]
betas.clean.tp3_heu_huu = betas.clean[,tp3_heu_huu]
table(pheno.tp3_heu_huu$Group)

```



Remove cpg sites from betas.clean that cause cpg.assoc errors.

```{r}
library(stringr)

#retrieving dataids for male and female samples.
pheno.tp3_heu_huu[which(predicted_sex=='m'),] %>% select(dataid) -> m5
pheno.tp3_heu_huu[which(predicted_sex=='f'),] %>% select(dataid) -> f5

##adding file/data location to align with clean.betas columns.
m5$file.loc = "C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\MICS Epigenetics Data\\MICS\\MICS\\All_raw_data"
f5$file.loc = "C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\MICS Epigenetics Data\\MICS\\MICS\\All_raw_data"

str_c(m5$file.loc, m5$dataid, sep = "\\") -> m5$betas.id ->m6
str_c(f5$file.loc, f5$dataid, sep = "\\") -> f5$betas.id ->f6

##creating vectors of male and female samples.
print(m6)
as.vector(m6)->m6
as.vector(f6)->f6

#Subset betas.clean for male and female samples. Remove sites for CpGs with all NAs for males or females.
betas.clean.new.m2 <- betas.clean.tp3_heu_huu[,m6]
betas.clean.new.f2 <- betas.clean.tp3_heu_huu[,f6]

remove.m2 <- which(rowSums(is.na(betas.clean.new.m2)) == length(m5$dataid))
remove.f2 <- which(rowSums(is.na(betas.clean.new.f2)) == length(f5$dataid))

betas.new.tp3_heu_huu <- betas.clean.tp3_heu_huu[-c(remove.m2,remove.f2),]

#total number of sites removed.
nrow(betas.clean.tp3_heu_huu) - nrow(betas.new.tp3_heu_huu)

```


Analyze the association between methylation beta values using betas.new and the phenotype of interest. Results show association between methylation beta values and Group column.

Set up a dummy variable that HEU=1, HUU=0

```{r}
pheno.tp3_heu_huu$hiv.cat= ifelse(pheno.tp3_heu_huu$Group == "HEU",1,0)

source("C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\Ziyi\\Code\\cpg.assoc_mod.R")

resultstp3_heu_huu.sex.race.ct5 = cpg.assoc_mod(betas.new.tp3_heu_huu,
                                                 pheno.tp3_heu_huu$hiv.cat,
                                                 covariates=as.data.frame(pheno.tp3_heu_huu[,.(predicted_sex,race_eth.new,CD8,CD4,NK,GR,MO)]
                                                 )
)
class(resultstp3_heu_huu.sex.race.ct5)
names(resultstp3_heu_huu.sex.race.ct5)
print(resultstp3_heu_huu.sex.race.ct5)
```

Check the top hits by ordering the p.value.
The effect size here is ~ mean difference in methylation proportion.
```{r}
head(cbind(resultstp3_heu_huu.sex.race.ct5$coefficients, P.value=resultstp3_heu_huu.sex.race.ct5$results[,3])[order(resultstp3_heu_huu.sex.race.ct5$results[,3]),])
```

Check Bonferroni significant sites of resultstp3_heu_huu.sex.race.ct5 using p-value/ncpg.

```{r}
table(resultstp3_heu_huu.sex.race.ct5$results[,3]< 0.05/nCpG)
```
#Genomic Inflation

Combine the sites with the manifest and view top sites. Use the combined data to create Volcano plot.

```{r}
datamanhattp3_heu_huu.sex.race.ct5 = cbind(resultstp3_heu_huu.sex.race.ct5$results, resultstp3_heu_huu.sex.race.ct5$coefficients)
setDT(datamanhattp3_heu_huu.sex.race.ct5)
datamanhattp3_heu_huu.sex.race.ct5 = datamanhattp3_heu_huu.sex.race.ct5[,.(probe_id=CPG.Labels,effect.size,std.error,P.value,FDR)]

datamanhattp3_heu_huu.sex.race.ct5 = merge(datamanhattp3_heu_huu.sex.race.ct5,manifest[,.(probe_id,chr,mapinfo)],by="probe_id")
datamanhattp3_heu_huu.sex.race.ct5[order(P.value)][1:10]
```

Volcano plot of resultstp3_heu_huu using Bonferroni threshold. 

2 sites met the Bonferroni threshold. 

#Volcano Plot
```{r}
with(datamanhattp3_heu_huu.sex.race.ct5, plot(effect.size, -log10(P.value), pch=20, main=""))
abline(h = -log10(0.05/(nCpG)), col = "blue", lty = 2, lwd = 1)
# abline(h = -log10(2.5e-07), col = "blue", lty = 2, lwd = 1)
abline(v = c(-0.05,0.05), col = "black", lty = 2, lwd = 1)

with(subset(datamanhattp3_heu_huu.sex.race.ct5, effect.size< -0.05 & FDR < 0.05), points(effect.size, -log10(P.value), pch=20, col="red"))

with(subset(datamanhattp3_heu_huu.sex.race.ct5, effect.size > 0.05 & FDR < 0.05), points(effect.size, -log10(P.value), pch=20, col="green"))
```

Cast the variable "chr" or chromosome to view significant sites by chromosome.

#Manhattan Plot
```{r}
datamanhattp3_heu_huu.sex.race.ct5[,chr:=as.integer(chr)]

qqman::manhattan(datamanhattp3_heu_huu.sex.race.ct5,chr="chr",bp="mapinfo",p="P.value",snp="probe_id"
                 ,suggestiveline=FALSE, genomewideline = -log10(0.05/(nCpG)),ylim=c(0,15)
                 ,main = "Manhattan Plot of mean DNA methylation diference\nbetween HEU and huu at Timepoint 3 - Adjusted")
```

#Limma

Use lmfit to fit a linear model to clean betas vs. huu status. Use fitted linear model to compute moderated t & F statistics, and log odds by empirical Bayes moderation of standard errors toward a global value.

```{r}
library(limma,minfi)

pheno.tp3_heu_huu$Group <- factor(pheno.tp3_heu_huu$Group, levels = c("HEU", "HUU"))
model = model.matrix( ~Group+predicted_sex+race_eth.new+CD8+CD4+NK+GR+MO,data=pheno.tp3_heu_huu)
EWAS.limma <- eBayes(lmFit(betas.new.tp3_heu_huu, design=model))
```

Check the top sites with statistically significant differential methylation across groups.

```{r}
Toptp3_heu_huu.sex.race.ct5<-topTable(EWAS.limma, coef=2, number=Inf, sort.by = "p")

head(Toptp3_heu_huu.sex.race.ct5)
```

Calculate the values of the two groups using getEWAP function. Generate dataframe called "average" to save top 500 sites for differential methylation between the two groups.

```{r}
genenames<-rownames(Toptp3_heu_huu.sex.race.ct5)
ewap<-getEAWP(betas.new.tp3_heu_huu)
average<- data.frame(heu=rep(0,500), huu=rep(0,500))
for (i in 1:1500){average[i,]<-tapply(ewap$exprs[genenames[i],], pheno.tp3_heu_huu$Group, mean, na.rm=TRUE)
}
average$CPG.Labels <- genenames[1:1500]
```

1. Use IlluminaHumanMethylationEPICanno.ilm10b4.hg19 to annotate the top sites with thier associated genes. 
2. Match TopTable with Annot output.
3. Remove unnecessary information.

```{r}
require(IlluminaHumanMethylationEPICanno.ilm10b4.hg19)

Annot<-as.data.frame(getAnnotation(IlluminaHumanMethylationEPICanno.ilm10b4.hg19))
Annot.Tops<- Annot[match(rownames(Toptp3_heu_huu.sex.race.ct5),Annot$Name),]
Annot.Tops<-Annot.Tops[,c("UCSC_RefGene_Name","UCSC_RefGene_Group","Relation_to_Island","chr","pos")]
Toptp3_heu_huu.sex.race.ct5<-cbind(Toptp3_heu_huu.sex.race.ct5, Annot.Tops)
head(Toptp3_heu_huu.sex.race.ct5)

```

Save results. Save logFC >= .05 sites as well as FDR, Bonferroni significant sites.

```{r}
Toptp3_heu_huu.sex.race.ct5 <- tibble::rownames_to_column(Toptp3_heu_huu.sex.race.ct5, "CPG.Labels")
Toptp3_heu_huu.sex.race.ct5 <- left_join(Toptp3_heu_huu.sex.race.ct5, resultstp3_heu_huu.sex.race.ct5$results, by = "CPG.Labels")

toptp3_heu_huu.sex.race.ct5 <- left_join(Toptp3_heu_huu.sex.race.ct5, average, by = "CPG.Labels")

```




## Code - Timepoint 12mo. - HIV vs. HEU, Sex & Race & ct3 (CD8, CD4, GR)

#Group - hiv vs. heu - Adjusted for race, sex, cell type 

Subset Timepoint 12 and compare group hiv & heu for both pheno and betas.clean. 66 samples were extracted.

hiv & heu at Timepoint 12 = 33 vs. 33.

```{r}
tp12_hiv_heu <- which(pheno$Timepoints==12 & pheno$Group!="HUU")
pheno.tp12_hiv_heu <- pheno[tp12_hiv_heu]
betas.clean.tp12_hiv_heu = betas.clean[,tp12_hiv_heu]
table(pheno.tp12_hiv_heu$Group)

```


Remove cpg sites from betas.clean that cause cpg.assoc errors.

```{r}
library(stringr)

#retrieving dataids for male and female samples.
pheno.tp12_hiv_heu[which(predicted_sex=='m'),"dataid"] -> m1
pheno.tp12_hiv_heu[which(predicted_sex=='f'),"dataid"] -> f1

  ##adding file/data location to align with clean.betas columns.
m1$file.loc = "C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\MICS Epigenetics Data\\MICS\\MICS\\All_raw_data"
f1$file.loc = "C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\MICS Epigenetics Data\\MICS\\MICS\\All_raw_data"

str_c(m1$file.loc, m1$dataid, sep = "\\") -> m1$betas.id ->m2
str_c(f1$file.loc, f1$dataid, sep = "\\") -> f1$betas.id ->f2
  
  ##creating vectors of male and female samples.
print(m2)
as.vector(m2)->m2
as.vector(f2)->f2

#Subset betas.clean for male and female samples. Remove sites for CpGs with all NAs for males or females.
betas.clean.new.m <- betas.clean.tp12_hiv_heu[,m2]
betas.clean.new.f <- betas.clean.tp12_hiv_heu[,f2]

remove.m <- which(rowSums(is.na(betas.clean.new.m)) == length(m1$dataid))
remove.f <- which(rowSums(is.na(betas.clean.new.f)) == length(f1$dataid))

betas.new.tp12_hiv_heu <- betas.clean.tp12_hiv_heu[-c(remove.m,remove.f),]

#total number of sites removed.
nrow(betas.clean.tp12_hiv_heu) - nrow(betas.new.tp12_hiv_heu)

```



Analyze the association between methylation beta values using betas.new and the phenotype of interest. Results show association between methylation beta values and Group column.

Set up a dummy variable that HIV+=1, HEU=0

```{r}
pheno.tp12_hiv_heu$hiv.cat= ifelse(pheno.tp12_hiv_heu$Group == "HIV+",1,0)

source("C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\Ziyi\\Code\\cpg.assoc_mod.R")

resultstp12_hiv_heu.sex.race.ct3 = cpg.assoc_mod(betas.new.tp12_hiv_heu,
                                                 pheno.tp12_hiv_heu$hiv.cat,
                                                 covariates=as.data.frame(pheno.tp12_hiv_heu[,.(predicted_sex,race_eth.new,CD8,CD4,GR)]
                                                 )
)
class(resultstp12_hiv_heu.sex.race.ct3)
names(resultstp12_hiv_heu.sex.race.ct3)
print(resultstp12_hiv_heu.sex.race.ct3)
```

Check the top hits by ordering the p.value.
The effect size here is ~ mean difference in methylation proportion.
```{r}
head(cbind(resultstp12_hiv_heu.sex.race.ct3$coefficients, P.value=resultstp12_hiv_heu.sex.race.ct3$results[,3])[order(resultstp12_hiv_heu.sex.race.ct3$results[,3]),])
```

Check Bonferroni significant sites of resultstp12_hiv_heu.sex.race.ct3 using p-value/ncpg.

```{r}
table(resultstp12_hiv_heu.sex.race.ct3$results[,3]< 0.05/nCpG)
```
#Genomic Inflation

Combine the sites with the manifest and view top sites. Use the combined data to create Volcano plot.

```{r}
datamanhattp12_hiv_heu.sex.race.ct3 = cbind(resultstp12_hiv_heu.sex.race.ct3$results, resultstp12_hiv_heu.sex.race.ct3$coefficients)
setDT(datamanhattp12_hiv_heu.sex.race.ct3)
datamanhattp12_hiv_heu.sex.race.ct3 = datamanhattp12_hiv_heu.sex.race.ct3[,.(probe_id=CPG.Labels,effect.size,std.error,P.value,FDR)]

datamanhattp12_hiv_heu.sex.race.ct3 = merge(datamanhattp12_hiv_heu.sex.race.ct3,manifest[,.(probe_id,chr,mapinfo)],by="probe_id")
datamanhattp12_hiv_heu.sex.race.ct3[order(P.value)][1:10]
```

Volcano plot of resultstp12_hiv_heu using Bonferroni threshold. 


#Volcano Plot
```{r}
with(datamanhattp12_hiv_heu.sex.race.ct3, plot(effect.size, -log10(P.value), pch=20, main=""))
abline(h = -log10(0.05/(nCpG)), col = "blue", lty = 2, lwd = 1)
# abline(h = -log10(2.5e-07), col = "blue", lty = 2, lwd = 1)
abline(v = c(-0.05,0.05), col = "black", lty = 2, lwd = 1)

with(subset(datamanhattp12_hiv_heu.sex.race.ct3, effect.size< -0.05 & FDR < 0.05), points(effect.size, -log10(P.value), pch=20, col="red"))

with(subset(datamanhattp12_hiv_heu.sex.race.ct3, effect.size > 0.05 & FDR < 0.05), points(effect.size, -log10(P.value), pch=20, col="green"))
```

Cast the variable "chr" or chromosome to view significant sites by chromosome.

#Manhattan Plot
```{r}
datamanhattp12_hiv_heu.sex.race.ct3[,chr:=as.integer(chr)]

qqman::manhattan(datamanhattp12_hiv_heu.sex.race.ct3,chr="chr",bp="mapinfo",p="P.value",snp="probe_id"
                 ,suggestiveline=FALSE, genomewideline = -log10(0.05/(nCpG)),ylim=c(0,15)
                 ,main = "Manhattan Plot of mean DNA methylation diference\nbetween HIV+ and HEU at Timepoint 12 - Adjusted")
```

#Limma

Use lmfit to fit a linear model to clean betas vs. heu status. Use fitted linear model to compute moderated t & F statistics, and log odds by empirical Bayes moderation of standard errors toward a global value.

```{r}
library(limma,minfi)

pheno.tp12_hiv_heu$Group <- factor(pheno.tp12_hiv_heu$Group, levels = c("HIV+", "HEU"))
model = model.matrix( ~Group+predicted_sex+race_eth.new+CD8+CD4+GR,data=pheno.tp12_hiv_heu)
EWAS.limma <- eBayes(lmFit(betas.new.tp12_hiv_heu, design=model))
```

Check the top sites with statistically significant differential methylation across groups.

```{r}
Toptp12_hiv_heu.sex.race.ct3<-topTable(EWAS.limma, coef=2, number=Inf, sort.by = "p")

head(Toptp12_hiv_heu.sex.race.ct3)
```

Calculate the values of the two groups using getEWAP function. Generate dataframe called "average" to save top 500 sites for differential methylation between the two groups.

```{r}
genenames<-rownames(Toptp12_hiv_heu.sex.race.ct3)
ewap<-getEAWP(betas.new.tp12_hiv_heu)
average<- data.frame(hiv=rep(0,500), heu=rep(0,500))
for (i in 1:1500){average[i,]<-tapply(ewap$exprs[genenames[i],], pheno.tp12_hiv_heu$Group, mean, na.rm=TRUE)
}
average$CPG.Labels <- genenames[1:1500]
```

1. Use IlluminaHumanMethylationEPICanno.ilm10b4.hg19 to annotate the top sites with thier associated genes. 
2. Match TopTable with Annot output.
3. Remove unnecessary information.

```{r}
require(IlluminaHumanMethylationEPICanno.ilm10b4.hg19)

Annot<-as.data.frame(getAnnotation(IlluminaHumanMethylationEPICanno.ilm10b4.hg19))
Annot.Tops<- Annot[match(rownames(Toptp12_hiv_heu.sex.race.ct3),Annot$Name),]
Annot.Tops<-Annot.Tops[,c("UCSC_RefGene_Name","UCSC_RefGene_Group","Relation_to_Island","chr","pos")]
Toptp12_hiv_heu.sex.race.ct3<-cbind(Toptp12_hiv_heu.sex.race.ct3, Annot.Tops)
head(Toptp12_hiv_heu.sex.race.ct3)

```

Save results. Save logFC >= .05 sites as well as FDR, Bonferroni significant sites.

```{r}
Toptp12_hiv_heu.sex.race.ct3 <- tibble::rownames_to_column(Toptp12_hiv_heu.sex.race.ct3, "CPG.Labels")
Toptp12_hiv_heu.sex.race.ct3 <- left_join(Toptp12_hiv_heu.sex.race.ct3, resultstp12_hiv_heu.sex.race.ct3$results, by = "CPG.Labels")

toptp12_hiv_heu.sex.race.ct3 <- left_join(Toptp12_hiv_heu.sex.race.ct3, average, by = "CPG.Labels")

```



## Code - Timepoint 12mo. - HIV vs. huu, Sex & Race & ct3 (CD8, CD4, GR)

#Group - hiv vs. huu - Adjusted for race, sex, cell type 

Subset Timepoint 12 and compare group hiv & huu for both pheno and betas.clean. 57 samples were extracted

hiv & huu at Timepoint 12 = 24 vs. 33.

```{r}
tp12_hiv_huu <- which(pheno$Timepoints==12 & pheno$Group!="HEU")
pheno.tp12_hiv_huu <- pheno[tp12_hiv_huu]
betas.clean.tp12_hiv_huu = betas.clean[,tp12_hiv_huu]
table(pheno.tp12_hiv_huu$Group)

```


Remove cpg sites from betas.clean that cause cpg.assoc errors.

```{r}
library(stringr)

#retrieving dataids for male and female samples.
pheno.tp12_hiv_huu[which(predicted_sex=='m'),"dataid"] -> m3
pheno.tp12_hiv_huu[which(predicted_sex=='f'), "dataid"] -> f3

  ##adding file/data location to align with clean.betas columns.
m3$file.loc = "C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\MICS Epigenetics Data\\MICS\\MICS\\All_raw_data"
f3$file.loc = "C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\MICS Epigenetics Data\\MICS\\MICS\\All_raw_data"

str_c(m3$file.loc, m3$dataid, sep = "\\") -> m3$betas.id ->m4
str_c(f3$file.loc, f3$dataid, sep = "\\") -> f3$betas.id ->f4
  
  ##creating vectors of male and female samples.
print(m4)
as.vector(m4)->m4
as.vector(f4)->f4

#Subset betas.clean for male and female samples. Remove sites for CpGs with all NAs for males or females.
betas.clean.new.m1 <- betas.clean.tp12_hiv_huu[,m4]
betas.clean.new.f1 <- betas.clean.tp12_hiv_huu[,f4]

remove.m1 <- which(rowSums(is.na(betas.clean.new.m1)) == length(m3$dataid))
remove.f1 <- which(rowSums(is.na(betas.clean.new.f1)) == length(f3$dataid))

betas.new.tp12_hiv_huu <- betas.clean.tp12_hiv_huu[-c(remove.m1,remove.f1),]

#total number of sites removed.
nrow(betas.clean.tp12_hiv_huu) - nrow(betas.new.tp12_hiv_huu)

```



Analyze the association between methylation beta values using betas.new and the phenotype of interest. Results show association between methylation beta values and Group column.

Set up a dummy variable that HIV+=1, HUU=0

```{r}
pheno.tp12_hiv_huu$hiv.cat= ifelse(pheno.tp12_hiv_huu$Group == "HIV+",1,0)

source("C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\Ziyi\\Code\\cpg.assoc_mod.R")

resultstp12_hiv_huu.sex.race.ct3 = cpg.assoc_mod(betas.new.tp12_hiv_huu,
                                                 pheno.tp12_hiv_huu$hiv.cat,
                                                 covariates=as.data.frame(pheno.tp12_hiv_huu[,.(predicted_sex,race_eth.new,CD8,CD4,GR)]
                                                 )
)
class(resultstp12_hiv_huu.sex.race.ct3)
names(resultstp12_hiv_huu.sex.race.ct3)
print(resultstp12_hiv_huu.sex.race.ct3)
```

Check the top hits by ordering the p.value.
The effect size here is ~ mean difference in methylation proportion.
```{r}
head(cbind(resultstp12_hiv_huu.sex.race.ct3$coefficients, P.value=resultstp12_hiv_huu.sex.race.ct3$results[,3])[order(resultstp12_hiv_huu.sex.race.ct3$results[,3]),])
```

Check Bonferroni significant sites of resultstp12_hiv_huu.sex.race.ct3 using p-value/ncpg.

```{r}
table(resultstp12_hiv_huu.sex.race.ct3$results[,3]< 0.05/nCpG)
```
#Genomic Inflation

Combine the sites with the manifest and view top sites. Use the combined data to create Volcano plot.

```{r}
datamanhattp12_hiv_huu.sex.race.ct3 = cbind(resultstp12_hiv_huu.sex.race.ct3$results, resultstp12_hiv_huu.sex.race.ct3$coefficients)
setDT(datamanhattp12_hiv_huu.sex.race.ct3)
datamanhattp12_hiv_huu.sex.race.ct3 = datamanhattp12_hiv_huu.sex.race.ct3[,.(probe_id=CPG.Labels,effect.size,std.error,P.value,FDR)]

datamanhattp12_hiv_huu.sex.race.ct3 = merge(datamanhattp12_hiv_huu.sex.race.ct3,manifest[,.(probe_id,chr,mapinfo)],by="probe_id")
datamanhattp12_hiv_huu.sex.race.ct3[order(P.value)][1:10]
```

Volcano plot of resultstp12_hiv_huu using Bonferroni threshold. 

2 sites met the Bonferroni threshold. 

#Volcano Plot
```{r}
with(datamanhattp12_hiv_huu.sex.race.ct3, plot(effect.size, -log10(P.value), pch=20, main=""))
abline(h = -log10(0.05/(nCpG)), col = "blue", lty = 2, lwd = 1)
# abline(h = -log10(2.5e-07), col = "blue", lty = 2, lwd = 1)
abline(v = c(-0.05,0.05), col = "black", lty = 2, lwd = 1)

with(subset(datamanhattp12_hiv_huu.sex.race.ct3, effect.size< -0.05 & FDR < 0.05), points(effect.size, -log10(P.value), pch=20, col="red"))

with(subset(datamanhattp12_hiv_huu.sex.race.ct3, effect.size > 0.05 & FDR < 0.05), points(effect.size, -log10(P.value), pch=20, col="green"))
```

Cast the variable "chr" or chromosome to view significant sites by chromosome.

#Manhattan Plot
```{r}
datamanhattp12_hiv_huu.sex.race.ct3[,chr:=as.integer(chr)]

qqman::manhattan(datamanhattp12_hiv_huu.sex.race.ct3,chr="chr",bp="mapinfo",p="P.value",snp="probe_id"
                 ,suggestiveline=FALSE, genomewideline = -log10(0.05/(nCpG)),ylim=c(0,15)
                 ,main = "Manhattan Plot of mean DNA methylation diference\nbetween HIV+ and huu at Timepoint 12 - Adjusted")
```

#Limma

Use lmfit to fit a linear model to clean betas vs. huu status. Use fitted linear model to compute moderated t & F statistics, and log odds by empirical Bayes moderation of standard errors toward a global value.

```{r}
library(limma,minfi)

pheno.tp12_hiv_huu$Group <- factor(pheno.tp12_hiv_huu$Group, levels = c("HIV+", "HUU"))
model = model.matrix( ~Group+predicted_sex+race_eth.new+CD8+CD4+GR,data=pheno.tp12_hiv_huu)
EWAS.limma <- eBayes(lmFit(betas.new.tp12_hiv_huu, design=model))
```

Check the top sites with statistically significant differential methylation across groups.

```{r}
Toptp12_hiv_huu.sex.race.ct3<-topTable(EWAS.limma, coef=2, number=Inf, sort.by = "p")

head(Toptp12_hiv_huu.sex.race.ct3)
```

Calculate the values of the two groups using getEWAP function. Generate dataframe called "average" to save top 500 sites for differential methylation between the two groups.

```{r}
genenames<-rownames(Toptp12_hiv_huu.sex.race.ct3)
ewap<-getEAWP(betas.new.tp12_hiv_huu)
average<- data.frame(hiv=rep(0,500), huu=rep(0,500))
for (i in 1:1500){average[i,]<-tapply(ewap$exprs[genenames[i],], pheno.tp12_hiv_huu$Group, mean, na.rm=TRUE)
}
average$CPG.Labels <- genenames[1:1500]
```

1. Use IlluminaHumanMethylationEPICanno.ilm10b4.hg19 to annotate the top sites with thier associated genes. 
2. Match TopTable with Annot output.
3. Remove unnecessary information.

```{r}
require(IlluminaHumanMethylationEPICanno.ilm10b4.hg19)

Annot<-as.data.frame(getAnnotation(IlluminaHumanMethylationEPICanno.ilm10b4.hg19))
Annot.Tops<- Annot[match(rownames(Toptp12_hiv_huu.sex.race.ct3),Annot$Name),]
Annot.Tops<-Annot.Tops[,c("UCSC_RefGene_Name","UCSC_RefGene_Group","Relation_to_Island","chr","pos")]
Toptp12_hiv_huu.sex.race.ct3<-cbind(Toptp12_hiv_huu.sex.race.ct3, Annot.Tops)
head(Toptp12_hiv_huu.sex.race.ct3)

```

Save results. 

```{r}
Toptp12_hiv_huu.sex.race.ct3 <- tibble::rownames_to_column(Toptp12_hiv_huu.sex.race.ct3, "CPG.Labels")
Toptp12_hiv_huu.sex.race.ct3 <- left_join(Toptp12_hiv_huu.sex.race.ct3, resultstp12_hiv_huu.sex.race.ct3$results, by = "CPG.Labels")

toptp12_hiv_huu.sex.race.ct3 <- left_join(Toptp12_hiv_huu.sex.race.ct3, average, by = "CPG.Labels")

```


## Code - timepoint 12mo. - Heu vs. huu, Sex & Race & ct3 (CD8, CD4, GR)

#Group - Heu vs. Huu - Adjusted for race, sex, cell type 

Subset timepoint 12 and compare group heu & huu for both pheno and betas.clean. 66 samples were extracted

heu & huu at timepoint 12 = 33 vs. 33.

```{r}
tp12_heu_huu <- which(pheno$Timepoints==12 & pheno$Group!="HIV+")
pheno.tp12_heu_huu <- pheno[tp12_heu_huu]
betas.clean.tp12_heu_huu = betas.clean[,tp12_heu_huu]
table(pheno.tp12_heu_huu$Group)

```


Remove cpg sites from betas.clean that cause cpg.assoc errors.

```{r}
library(stringr)

#retrieving dataids for male and female samples.
pheno.tp12_heu_huu[which(predicted_sex=='m'),] %>% select(dataid) -> m5
pheno.tp12_heu_huu[which(predicted_sex=='f'),] %>% select(dataid) -> f5

  ##adding file/data location to align with clean.betas columns.
m5$file.loc = "C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\MICS Epigenetics Data\\MICS\\MICS\\All_raw_data"
f5$file.loc = "C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\MICS Epigenetics Data\\MICS\\MICS\\All_raw_data"

str_c(m5$file.loc, m5$dataid, sep = "\\") -> m5$betas.id ->m6
str_c(f5$file.loc, f5$dataid, sep = "\\") -> f5$betas.id ->f6
  
  ##creating vectors of male and female samples.
print(m6)
as.vector(m6)->m6
as.vector(f6)->f6

#Subset betas.clean for male and female samples. Remove sites for CpGs with all NAs for males or females.
betas.clean.new.m2 <- betas.clean.tp12_heu_huu[,m6]
betas.clean.new.f2 <- betas.clean.tp12_heu_huu[,f6]

remove.m2 <- which(rowSums(is.na(betas.clean.new.m2)) == length(m5$dataid))
remove.f2 <- which(rowSums(is.na(betas.clean.new.f2)) == length(f5$dataid))

betas.new.tp12_heu_huu <- betas.clean.tp12_heu_huu[-c(remove.m2,remove.f2),]

#total number of sites removed.
nrow(betas.clean.tp12_heu_huu) - nrow(betas.new.tp12_heu_huu)

```


Analyze the association between methylation beta values using betas.new and the phenotype of interest. Results show association between methylation beta values and Group column.

Set up a dummy variable that HEU=1, HUU=0

```{r}
pheno.tp12_heu_huu$hiv.cat= ifelse(pheno.tp12_heu_huu$Group == "HEU",1,0)

source("C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\Ziyi\\Code\\cpg.assoc_mod.R")

resultstp12_heu_huu.sex.race.ct3 = cpg.assoc_mod(betas.new.tp12_heu_huu,
                                                 pheno.tp12_heu_huu$hiv.cat,
                                                 covariates=as.data.frame(pheno.tp12_heu_huu[,.(predicted_sex,race_eth.new,CD8,CD4,GR)]
                                                 )
)
class(resultstp12_heu_huu.sex.race.ct3)
names(resultstp12_heu_huu.sex.race.ct3)
print(resultstp12_heu_huu.sex.race.ct3)
```

Check the top hits by ordering the p.value.
The effect size here is ~ mean difference in methylation proportion.
```{r}
head(cbind(resultstp12_heu_huu.sex.race.ct3$coefficients, P.value=resultstp12_heu_huu.sex.race.ct3$results[,3])[order(resultstp12_heu_huu.sex.race.ct3$results[,3]),])
```

Check Bonferroni significant sites of resultstp12_heu_huu.sex.race.ct3 using p-value/ncpg.

```{r}
table(resultstp12_heu_huu.sex.race.ct3$results[,3]< 0.05/nCpG)
```
#Genomic Inflation

Combine the sites with the manifest and view top sites. Use the combined data to create Volcano plot.

```{r}
datamanhattp12_heu_huu.sex.race.ct3 = cbind(resultstp12_heu_huu.sex.race.ct3$results, resultstp12_heu_huu.sex.race.ct3$coefficients)
setDT(datamanhattp12_heu_huu.sex.race.ct3)
datamanhattp12_heu_huu.sex.race.ct3 = datamanhattp12_heu_huu.sex.race.ct3[,.(probe_id=CPG.Labels,effect.size,std.error,P.value,FDR)]

datamanhattp12_heu_huu.sex.race.ct3 = merge(datamanhattp12_heu_huu.sex.race.ct3,manifest[,.(probe_id,chr,mapinfo)],by="probe_id")
datamanhattp12_heu_huu.sex.race.ct3[order(P.value)][1:10]
```

Volcano plot of resultstp12_heu_huu using Bonferroni threshold. 

#Volcano Plot
```{r}
with(datamanhattp12_heu_huu.sex.race.ct3, plot(effect.size, -log10(P.value), pch=20, main="", xlim= c(-.5,.5)))
abline(h = -log10(0.05/(nCpG)), col = "blue", lty = 2, lwd = 1)
# abline(h = -log10(2.5e-07), col = "blue", lty = 2, lwd = 1)
abline(v = c(-0.05,0.05), col = "black", lty = 2, lwd = 1)

with(subset(datamanhattp12_heu_huu.sex.race.ct3, effect.size< -0.05 & FDR < 0.05), points(effect.size, -log10(P.value), pch=20, col="red"))

with(subset(datamanhattp12_heu_huu.sex.race.ct3, effect.size > 0.05 & FDR < 0.05), points(effect.size, -log10(P.value), pch=20, col="green"))
```

Cast the variable "chr" or chromosome to view significant sites by chromosome.

#Manhattan Plot
```{r}
datamanhattp12_heu_huu.sex.race.ct3[,chr:=as.integer(chr)]

qqman::manhattan(datamanhattp12_heu_huu.sex.race.ct3,chr="chr",bp="mapinfo",p="P.value",snp="probe_id"
                 ,suggestiveline=FALSE, genomewideline = -log10(0.05/(nCpG)),ylim=c(0,15)
                 ,main = "Manhattan Plot of mean DNA methylation diference\nbetween HEU and huu at timepoint 12 - Adjusted")
```

#Limma

Use lmfit to fit a linear model to clean betas vs. huu status. Use fitted linear model to compute moderated t & F statistics, and log odds by empirical Bayes moderation of standard errors toward a global value.

```{r}
library(limma,minfi)

pheno.tp12_heu_huu$Group <- factor(pheno.tp12_heu_huu$Group, levels = c("HEU", "HUU"))
model = model.matrix( ~Group+predicted_sex+race_eth.new+CD8+CD4+GR,data=pheno.tp12_heu_huu)
EWAS.limma <- eBayes(lmFit(betas.new.tp12_heu_huu, design=model))
```

Check the top sites with statistically significant differential methylation across groups.

```{r}
Toptp12_heu_huu.sex.race.ct3<-topTable(EWAS.limma, coef=2, number=Inf, sort.by = "p")

head(Toptp12_heu_huu.sex.race.ct3)
```

Calculate the values of the two groups using getEWAP function. Generate dataframe called "average" to save top 500 sites for differential methylation between the two groups.

```{r}
genenames<-rownames(Toptp12_heu_huu.sex.race.ct3)
ewap<-getEAWP(betas.new.tp12_heu_huu)
average<- data.frame(heu=rep(0,500), huu=rep(0,500))
for (i in 1:1500){average[i,]<-tapply(ewap$exprs[genenames[i],], pheno.tp12_heu_huu$Group, mean, na.rm=TRUE)
}
average$CPG.Labels <- genenames[1:1500]
```

1. Use IlluminaHumanMethylationEPICanno.ilm10b4.hg19 to annotate the top sites with thier associated genes. 
2. Match TopTable with Annot output.
3. Remove unnecessary information.

```{r}
require(IlluminaHumanMethylationEPICanno.ilm10b4.hg19)

Annot<-as.data.frame(getAnnotation(IlluminaHumanMethylationEPICanno.ilm10b4.hg19))
Annot.Tops<- Annot[match(rownames(Toptp12_heu_huu.sex.race.ct3),Annot$Name),]
Annot.Tops<-Annot.Tops[,c("UCSC_RefGene_Name","UCSC_RefGene_Group","Relation_to_Island","chr","pos")]
Toptp12_heu_huu.sex.race.ct3<-cbind(Toptp12_heu_huu.sex.race.ct3, Annot.Tops)
head(Toptp12_heu_huu.sex.race.ct3)

```

Save results. 

```{r}
Toptp12_heu_huu.sex.race.ct3 <- tibble::rownames_to_column(Toptp12_heu_huu.sex.race.ct3, "CPG.Labels")
Toptp12_heu_huu.sex.race.ct3 <- left_join(Toptp12_heu_huu.sex.race.ct3, resultstp12_heu_huu.sex.race.ct3$results, by = "CPG.Labels")

toptp12_heu_huu.sex.race.ct3 <- left_join(Toptp12_heu_huu.sex.race.ct3, average, by = "CPG.Labels")

```




#Targeted Gene Analysis - Adjusted for Sex, Race, and Cell type


#EBF4 - Timepoint 3 - HIV vs. HEU

Target the EBF4 gene. Timepoint 3

Note: if logFC > 0, means HIV < HEU.


```{r}
Toptp3_hiv_heu.sex.race.ct5.EBF4 <- Toptp3_hiv_heu.sex.race.ct5[grepl("EBF4", Toptp3_hiv_heu.sex.race.ct5$UCSC_RefGene_Name),]

table(Toptp3_hiv_heu.sex.race.ct5.EBF4[,14] < 0.05/(56))

```

0 sites were found.

```{r}
Toptp3_hiv_heu.sex.race.ct5.EBF4.bonf <- Toptp3_hiv_heu.sex.race.ct5.EBF4[which(Toptp3_hiv_heu.sex.race.ct5.EBF4$P.value < 0.05/(56)),]

cpg_hist <- Toptp3_hiv_heu.sex.race.ct5.EBF4.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with EBF4 \n between HIV+ and HEU group at timepoint 3")
```


#XDH
Timepoint 3
HIV vs. HEU
Target the XDH gene. Timepoint 3

```{r}
Toptp3_hiv_heu.sex.race.ct5.XDH <- Toptp3_hiv_heu.sex.race.ct5[grepl("XDH", Toptp3_hiv_heu.sex.race.ct5$UCSC_RefGene_Name),]

table(Toptp3_hiv_heu.sex.race.ct5.XDH[,14] < 0.05/(21))
```

0 sites were found.

```{r}
Toptp3_hiv_heu.sex.race.ct5.XDH.bonf <- Toptp3_hiv_heu.sex.race.ct5.XDH[which(Toptp3_hiv_heu.sex.race.ct5.XDH$P.value < 0.05/(21)),]

cpg_hist <- Toptp3_hiv_heu.sex.race.ct5.XDH.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with XDH \n between HIV+ and HEU group at timepoint 3")

```



SPERT
Timepoint 3
HIV vs. HEU
Target the SPERT gene. Timepoint 3


```{r}
Toptp3_hiv_heu.sex.race.ct5.SPERT <- Toptp3_hiv_heu.sex.race.ct5[grepl("SPERT", Toptp3_hiv_heu.sex.race.ct5$UCSC_RefGene_Name),]

table(Toptp3_hiv_heu.sex.race.ct5.SPERT[,14] < 0.05/(17))
```

0 sites were found.

```{r}
Toptp3_hiv_heu.sex.race.ct5.SPERT.bonf <- Toptp3_hiv_heu.sex.race.ct5.SPERT[which(Toptp3_hiv_heu.sex.race.ct5.SPERT$P.value < 0.05/(17)),]

cpg_hist <- Toptp3_hiv_heu.sex.race.ct5.SPERT.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with SPERT \n between HIV+ and HEU group at timepoint 3")
```


NLRC5
Timepoint 3
HIV vs. HEU
Target the NLRC5 gene. Timepoint 3

```{r}
Toptp3_hiv_heu.sex.race.ct5.NLRC5 <- Toptp3_hiv_heu.sex.race.ct5[grepl("NLRC5", Toptp3_hiv_heu.sex.race.ct5$UCSC_RefGene_Name),]

table(Toptp3_hiv_heu.sex.race.ct5.NLRC5[,14] < 0.05/(62))
```

0 sites were found.

```{r}
Toptp3_hiv_heu.sex.race.ct5.NLRC5.bonf <- Toptp3_hiv_heu.sex.race.ct5.NLRC5[which(Toptp3_hiv_heu.sex.race.ct5.NLRC5$P.value < 0.05/(62)),]

cpg_hist <- Toptp3_hiv_heu.sex.race.ct5.NLRC5.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with NLRC5 \n between HIV+ and HEU group at timepoint 3")

```


FOXP1
Timepoint 3
HIV vs. HEU
Target the FOXP1 gene. Timepoint 3

```{r}
Toptp3_hiv_heu.sex.race.ct5.FOXP1 <- Toptp3_hiv_heu.sex.race.ct5[grepl("FOXP1", Toptp3_hiv_heu.sex.race.ct5$UCSC_RefGene_Name),]

table(Toptp3_hiv_heu.sex.race.ct5.FOXP1[,14] < 0.05/(299))
```

0 sites were found.

```{r}
Toptp3_hiv_heu.sex.race.ct5.FOXP1.bonf <- Toptp3_hiv_heu.sex.race.ct5.FOXP1[which(Toptp3_hiv_heu.sex.race.ct5.FOXP1$P.value < 0.05/(299)),]

cpg_hist <- Toptp3_hiv_heu.sex.race.ct5.FOXP1.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with FOXP1 \n between HIV+ and HEU group at timepoint 3")

```



FNDC3B
Timepoint 3
HIV vs. HEU
Target the FNDC3B gene. Timepoint 3

```{r}
Toptp3_hiv_heu.sex.race.ct5.FNDC3B <- Toptp3_hiv_heu.sex.race.ct5[grepl("FNDC3B", Toptp3_hiv_heu.sex.race.ct5$UCSC_RefGene_Name),]

table(Toptp3_hiv_heu.sex.race.ct5.FNDC3B[,14] < 0.05/(152))
```

0 sites were found.
```{r}
Toptp3_hiv_heu.sex.race.ct5.FNDC3B.bonf <- Toptp3_hiv_heu.sex.race.ct5.FNDC3B[which(Toptp3_hiv_heu.sex.race.ct5.FNDC3B$P.value < 0.05/(152)),]

cpg_hist <- Toptp3_hiv_heu.sex.race.ct5.FNDC3B.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with FNDC3B \n between HIV+ and HEU group at timepoint 3")
```


RPS6KA2
Timepoint 3
HIV vs. HEU
Target the RPS6KA2 gene. Timepoint 3

```{r}
Toptp3_hiv_heu.sex.race.ct5.RPS6KA2 <- Toptp3_hiv_heu.sex.race.ct5[grepl("RPS6KA2", Toptp3_hiv_heu.sex.race.ct5$UCSC_RefGene_Name),]

table(Toptp3_hiv_heu.sex.race.ct5.RPS6KA2[,14] < 0.05/(362))
```

0 sites were found.
```{r}
Toptp3_hiv_heu.sex.race.ct5.RPS6KA2.bonf <- Toptp3_hiv_heu.sex.race.ct5.RPS6KA2[which(Toptp3_hiv_heu.sex.race.ct5.RPS6KA2$P.value < 0.05/(362)),]

cpg_hist <- Toptp3_hiv_heu.sex.race.ct5.RPS6KA2.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with RPS6KA2 \n between HIV+ and HEU group at timepoint 3")

```



DLL1
Timepoint 3
HIV vs. HEU
Target the DLL1 gene. Timepoint 3

```{r}
Toptp3_hiv_heu.sex.race.ct5.DLL1 <- Toptp3_hiv_heu.sex.race.ct5[grepl("DLL1", Toptp3_hiv_heu.sex.race.ct5$UCSC_RefGene_Name),]

table(Toptp3_hiv_heu.sex.race.ct5.DLL1[,14] < 0.05/(38))
```

0 site was found.
```{r}
Toptp3_hiv_heu.sex.race.ct5.DLL1.bonf <- Toptp3_hiv_heu.sex.race.ct5.DLL1[which(Toptp3_hiv_heu.sex.race.ct5.DLL1$P.value < 0.05/(38)),]

cpg_hist <- Toptp3_hiv_heu.sex.race.ct5.DLL1.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with DLL1 \n between HIV+ and HEU group at timepoint 3")

```


HCG22
Timepoint 3
HIV vs. HEU
Target the HCG22 gene. Timepoint 3

```{r}
Toptp3_hiv_heu.sex.race.ct5.HCG22 <- Toptp3_hiv_heu.sex.race.ct5[grepl("HCG22", Toptp3_hiv_heu.sex.race.ct5$UCSC_RefGene_Name),]

table(Toptp3_hiv_heu.sex.race.ct5.HCG22[,14] < 0.05/(30))
```

0 sites were found.
```{r}
Toptp3_hiv_heu.sex.race.ct5.HCG22.bonf <- Toptp3_hiv_heu.sex.race.ct5.HCG22[which(Toptp3_hiv_heu.sex.race.ct5.HCG22$P.value < 0.05/(30)),]

cpg_hist <- Toptp3_hiv_heu.sex.race.ct5.HCG22.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with HCG22 \n between HIV+ and HEU group at timepoint 3")
```


#EBF4 - Timepoint 3 - HIV vs. HUU
Target the EBF4 gene. Timepoint 3


```{r}
Toptp3_hiv_huu.sex.race.ct5.EBF4 <- Toptp3_hiv_huu.sex.race.ct5[grepl("EBF4", Toptp3_hiv_huu.sex.race.ct5$UCSC_RefGene_Name),]
table(Toptp3_hiv_huu.sex.race.ct5.EBF4[,14] < 0.05/(56))
```

0 sites were found.

```{r}
Toptp3_hiv_huu.sex.race.ct5.EBF4.bonf <- Toptp3_hiv_huu.sex.race.ct5.EBF4[which(Toptp3_hiv_huu.sex.race.ct5.EBF4$P.value < 0.05/(56)),]

cpg_hist <- Toptp3_hiv_huu.sex.race.ct5.EBF4.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with EBF4 \n between HIV+ and HUU group at timepoint 3")
```

XDH
Timepoint 3
HIV vs. HUU
Target the XDH gene. Timepoint 3

```{r}
Toptp3_hiv_huu.sex.race.ct5.XDH <- Toptp3_hiv_huu.sex.race.ct5[grepl("XDH", Toptp3_hiv_huu.sex.race.ct5$UCSC_RefGene_Name),]

table(Toptp3_hiv_huu.sex.race.ct5.XDH[,14] < 0.05/(21))
```

0 sites were found.
```{r}
Toptp3_hiv_huu.sex.race.ct5.XDH.bonf <- Toptp3_hiv_huu.sex.race.ct5.XDH[which(Toptp3_hiv_huu.sex.race.ct5.XDH$P.value < 0.05/(21)),]

cpg_hist <- Toptp3_hiv_huu.sex.race.ct5.XDH.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with XDH \n between HIV+ and HUU group at timepoint 3")
```


SPERT
Timepoint 3
HIV vs. HUU
Target the SPERT gene. Timepoint 3

```{r}
Toptp3_hiv_huu.sex.race.ct5.SPERT <- Toptp3_hiv_huu.sex.race.ct5[grepl("SPERT", Toptp3_hiv_huu.sex.race.ct5$UCSC_RefGene_Name),]

table(Toptp3_hiv_huu.sex.race.ct5.SPERT[,14] < 0.05/(17))
```

0 sites were found.

```{r}
Toptp3_hiv_huu.sex.race.ct5.SPERT.bonf <- Toptp3_hiv_huu.sex.race.ct5.SPERT[which(Toptp3_hiv_huu.sex.race.ct5.SPERT$P.value < 0.05/(17)),]

cpg_hist <- Toptp3_hiv_huu.sex.race.ct5.SPERT.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with SPERT \n between HIV+ and HUU group at timepoint 3")
```


NLRC5
Timepoint 3
HIV vs. HUU
Target the NLRC5 gene. Timepoint 3

```{r}
Toptp3_hiv_huu.sex.race.ct5.NLRC5 <- Toptp3_hiv_huu.sex.race.ct5[grepl("NLRC5", Toptp3_hiv_huu.sex.race.ct5$UCSC_RefGene_Name),]

table(Toptp3_hiv_huu.sex.race.ct5.NLRC5[,14] < 0.05/(62))
```

0 sites were found.

```{r}
Toptp3_hiv_huu.sex.race.ct5.NLRC5.bonf <- Toptp3_hiv_huu.sex.race.ct5.NLRC5[which(Toptp3_hiv_huu.sex.race.ct5.NLRC5$P.value < 0.05/(62)),]

cpg_hist <- Toptp3_hiv_huu.sex.race.ct5.NLRC5.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with NLRC5 \n between HIV+ and HUU group at timepoint 3")
```


FOXP1
Timepoint 3
HIV vs. HUU
Target the FOXP1 gene. Timepoint 3


```{r}
Toptp3_hiv_huu.sex.race.ct5.FOXP1 <- Toptp3_hiv_huu.sex.race.ct5[grepl("FOXP1", Toptp3_hiv_huu.sex.race.ct5$UCSC_RefGene_Name),]

table(Toptp3_hiv_huu.sex.race.ct5.FOXP1[,14] < 0.05/(299))
```

0 were found. 

```{r}
Toptp3_hiv_huu.sex.race.ct5.FOXP1.bonf <- Toptp3_hiv_huu.sex.race.ct5.FOXP1[which(Toptp3_hiv_huu.sex.race.ct5.FOXP1$P.value < 0.05/(299)),]

cpg_hist <- Toptp3_hiv_huu.sex.race.ct5.FOXP1.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with FOXP1 \n between HIV+ and HUU group at timepoint 3")

```

FNDC3B
Timepoint 3
HIV vs. HUU
Target the FNDC3B gene. Timepoint 3


```{r}
Toptp3_hiv_huu.sex.race.ct5.FNDC3B <- Toptp3_hiv_huu.sex.race.ct5[grepl("FNDC3B", Toptp3_hiv_huu.sex.race.ct5$UCSC_RefGene_Name),]

table(Toptp3_hiv_huu.sex.race.ct5.FNDC3B[,14] < 0.05/(152))
```

0 sites were found.

```{r}
Toptp3_hiv_huu.sex.race.ct5.FNDC3B.bonf <- Toptp3_hiv_huu.sex.race.ct5.FNDC3B[which(Toptp3_hiv_huu.sex.race.ct5.FNDC3B$P.value < 0.05/(152)),]

cpg_hist <- Toptp3_hiv_huu.sex.race.ct5.FNDC3B.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with FNDC3B \n between HIV+ and HUU group at timepoint 3")
```

RPS6KA2
Timepoint 3
HIV vs. HUU
Target the RPS6KA2 gene. Timepoint 3

```{r}
Toptp3_hiv_huu.sex.race.ct5.RPS6KA2 <- Toptp3_hiv_huu.sex.race.ct5[grepl("RPS6KA2", Toptp3_hiv_huu.sex.race.ct5$UCSC_RefGene_Name),]

table(Toptp3_hiv_huu.sex.race.ct5.RPS6KA2[,14] < 0.05/(362))
```

0 sites were found.

```{r}
Toptp3_hiv_huu.sex.race.ct5.RPS6KA2.bonf <- Toptp3_hiv_huu.sex.race.ct5.RPS6KA2[which(Toptp3_hiv_huu.sex.race.ct5.RPS6KA2$P.value < 0.05/(362)),]

cpg_hist <- Toptp3_hiv_huu.sex.race.ct5.RPS6KA2.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with RPS6KA2 \n between HIV+ and HUU group at timepoint 3")
```

DLL1
Timepoint 3
HIV vs. HUU
Target the DLL1 gene. Timepoint 3

```{r}
Toptp3_hiv_huu.sex.race.ct5.DLL1 <- Toptp3_hiv_huu.sex.race.ct5[grepl("DLL1", Toptp3_hiv_huu.sex.race.ct5$UCSC_RefGene_Name),]

table(Toptp3_hiv_huu.sex.race.ct5.DLL1[,14] < 0.05/(38))
```
 0 sites were found.

```{r}
Toptp3_hiv_huu.sex.race.ct5.DLL1.bonf <- Toptp3_hiv_huu.sex.race.ct5.DLL1[which(Toptp3_hiv_huu.sex.race.ct5.DLL1$P.value < 0.05/(38)),]

cpg_hist <- Toptp3_hiv_huu.sex.race.ct5.DLL1.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with DLL1 \n between HIV+ and HUU group at timepoint 3")
```

HCG22
Timepoint 3
HIV vs. HUU
Target the HCG22 gene. Timepoint 3

```{r}
Toptp3_hiv_huu.sex.race.ct5.HCG22 <- Toptp3_hiv_huu.sex.race.ct5[grepl("HCG22", Toptp3_hiv_huu.sex.race.ct5$UCSC_RefGene_Name),]

table(Toptp3_hiv_huu.sex.race.ct5.HCG22[,14] < 0.05/(30))
```

0 sites were found.

```{r}
Toptp3_hiv_huu.sex.race.ct5.HCG22.bonf <- Toptp3_hiv_huu.sex.race.ct5.HCG22[which(Toptp3_hiv_huu.sex.race.ct5.HCG22$P.value < 0.05/(30)),]

cpg_hist <- Toptp3_hiv_huu.sex.race.ct5.HCG22.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with HCG22 \n between HIV+ and HUU group at timepoint 3")
```



#EBF4 - Timepoint 3 - HEU vs. HUU
Target the EBF4 gene. Timepoint 3

Note: if logFC > 0, means heu < huu.

```{r}
Toptp3_heu_huu.sex.race.ct5.EBF4 <- Toptp3_heu_huu.sex.race.ct5[grepl("EBF4", Toptp3_heu_huu.sex.race.ct5$UCSC_RefGene_Name),]
table(Toptp3_heu_huu.sex.race.ct5.EBF4[,14] < 0.05/(56))
```

0 sites were found.

```{r}
Toptp3_heu_huu.sex.race.ct5.EBF4.bonf <- Toptp3_heu_huu.sex.race.ct5.EBF4[which(Toptp3_heu_huu.sex.race.ct5.EBF4$P.value < 0.05/(56)),]

cpg_hist <- Toptp3_heu_huu.sex.race.ct5.EBF4.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with EBF4 \n between heu+ and HUU group at timepoint 3")
```


XDH
Timepoint 3
heu vs. HUU
Target the XDH gene. Timepoint 3

```{r}
Toptp3_heu_huu.sex.race.ct5.XDH <- Toptp3_heu_huu.sex.race.ct5[grepl("XDH", Toptp3_heu_huu.sex.race.ct5$UCSC_RefGene_Name),]

table(Toptp3_heu_huu.sex.race.ct5.XDH[,14] < 0.05/(21))
```

1 sites were found.

```{r}
Toptp3_heu_huu.sex.race.ct5.XDH.bonf <- Toptp3_heu_huu.sex.race.ct5.XDH[which(Toptp3_heu_huu.sex.race.ct5.XDH$P.value < 0.05/(21)),]

cpg_hist <- Toptp3_heu_huu.sex.race.ct5.XDH.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with XDH \n between heu+ and HUU group at timepoint 3")
```

Save results.

```{r}
genename.XDH <- Toptp3_heu_huu.sex.race.ct5.XDH$CPG.Labels
average.XDH <- data.frame(heu = rep(0, length(genename.XDH)), HUU = rep(0, length(genename.XDH)))
eawp <- getEAWP(betas.new.tp3_heu_huu)
for (i in 1:length(genename.XDH)){
  average.XDH[i,] <- tapply(eawp$exprs[genename.XDH[i],], pheno.tp3_heu_huu$Group, mean, na.rm=TRUE)
}
average.XDH$CPG.Labels <- genename.XDH
toptp3_heu_huu.sex.race.ct5.XDH <- left_join(Toptp3_heu_huu.sex.race.ct5.XDH, average.XDH, by="CPG.Labels")

write.csv(toptp3_heu_huu.sex.race.ct5.XDH, file = "C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\Jasmine\\Aim 1\\toptp3_heu_huu.sex.race.ct5.XDH.csv")
```



SPERT
Timepoint 3
heu vs. HUU
Target the SPERT gene. Timepoint 3

```{r}
Toptp3_heu_huu.sex.race.ct5.SPERT <- Toptp3_heu_huu.sex.race.ct5[grepl("SPERT", Toptp3_heu_huu.sex.race.ct5$UCSC_RefGene_Name),]

table(Toptp3_heu_huu.sex.race.ct5.SPERT[,14] < 0.05/(17))
```

0 sites were found.

```{r}
Toptp3_heu_huu.sex.race.ct5.SPERT.bonf <- Toptp3_heu_huu.sex.race.ct5.SPERT[which(Toptp3_heu_huu.sex.race.ct5.SPERT$P.value < 0.05/(17)),]

cpg_hist <- Toptp3_heu_huu.sex.race.ct5.SPERT.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with SPERT \n between heu+ and HUU group at timepoint 3")
```


NLRC5
Timepoint 3
heu vs. HUU
Target the NLRC5 gene. Timepoint 3

```{r}
Toptp3_heu_huu.sex.race.ct5.NLRC5 <- Toptp3_heu_huu.sex.race.ct5[grepl("NLRC5", Toptp3_heu_huu.sex.race.ct5$UCSC_RefGene_Name),]

table(Toptp3_heu_huu.sex.race.ct5.NLRC5[,14] < 0.05/(62))
```

0 sites were found.

```{r}
Toptp3_heu_huu.sex.race.ct5.NLRC5.bonf <- Toptp3_heu_huu.sex.race.ct5.NLRC5[which(Toptp3_heu_huu.sex.race.ct5.NLRC5$P.value < 0.05/(62)),]

cpg_hist <- Toptp3_heu_huu.sex.race.ct5.NLRC5.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with NLRC5 \n between heu+ and HUU group at timepoint 3")
```

FOXP1
Timepoint 3
heu vs. HUU
Target the FOXP1 gene. Timepoint 3


```{r}
Toptp3_heu_huu.sex.race.ct5.FOXP1 <- Toptp3_heu_huu.sex.race.ct5[grepl("FOXP1", Toptp3_heu_huu.sex.race.ct5$UCSC_RefGene_Name),]

table(Toptp3_heu_huu.sex.race.ct5.FOXP1[,14] < 0.05/(299))
```

0 sites were found.

```{r}
Toptp3_heu_huu.sex.race.ct5.FOXP1.bonf <- Toptp3_heu_huu.sex.race.ct5.FOXP1[which(Toptp3_heu_huu.sex.race.ct5.FOXP1$P.value < 0.05/(299)),]

cpg_hist <- Toptp3_heu_huu.sex.race.ct5.FOXP1.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with FOXP1 \n between heu+ and HUU group at timepoint 3")

```


FNDC3B
Timepoint 3
heu vs. HUU
Target the FNDC3B gene. Timepoint 3


```{r}
Toptp3_heu_huu.sex.race.ct5.FNDC3B <- Toptp3_heu_huu.sex.race.ct5[grepl("FNDC3B", Toptp3_heu_huu.sex.race.ct5$UCSC_RefGene_Name),]

table(Toptp3_heu_huu.sex.race.ct5.FNDC3B[,14] < 0.05/(152))
```

0 sites were found.

```{r}
Toptp3_heu_huu.sex.race.ct5.FNDC3B.bonf <- Toptp3_heu_huu.sex.race.ct5.FNDC3B[which(Toptp3_heu_huu.sex.race.ct5.FNDC3B$P.value < 0.05/(152)),]

cpg_hist <- Toptp3_heu_huu.sex.race.ct5.FNDC3B.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with FNDC3B \n between heu+ and HUU group at timepoint 3")
```


RPS6KA2
Timepoint 3
heu vs. HUU
Target the RPS6KA2 gene. Timepoint 3

```{r}
Toptp3_heu_huu.sex.race.ct5.RPS6KA2 <- Toptp3_heu_huu.sex.race.ct5[grepl("RPS6KA2", Toptp3_heu_huu.sex.race.ct5$UCSC_RefGene_Name),]

table(Toptp3_heu_huu.sex.race.ct5.RPS6KA2[,14] < 0.05/(362))
```

0 sites were found.

```{r}
Toptp3_heu_huu.sex.race.ct5.RPS6KA2.bonf <- Toptp3_heu_huu.sex.race.ct5.RPS6KA2[which(Toptp3_heu_huu.sex.race.ct5.RPS6KA2$P.value < 0.05/(362)),]

cpg_hist <- Toptp3_heu_huu.sex.race.ct5.RPS6KA2.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with RPS6KA2 \n between heu+ and HUU group at timepoint 3")
```


DLL1
Timepoint 3
heu vs. HUU
Target the DLL1 gene. Timepoint 3

```{r}
Toptp3_heu_huu.sex.race.ct5.DLL1 <- Toptp3_heu_huu.sex.race.ct5[grepl("DLL1", Toptp3_heu_huu.sex.race.ct5$UCSC_RefGene_Name),]

table(Toptp3_heu_huu.sex.race.ct5.DLL1[,14] < 0.05/(38))
```

0 sites were found.

```{r}
Toptp3_heu_huu.sex.race.ct5.DLL1.bonf <- Toptp3_heu_huu.sex.race.ct5.DLL1[which(Toptp3_heu_huu.sex.race.ct5.DLL1$P.value < 0.05/(38)),]

cpg_hist <- Toptp3_heu_huu.sex.race.ct5.DLL1.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with DLL1 \n between heu+ and HUU group at timepoint 3")
```


HCG22
Timepoint 3
heu vs. HUU
Target the HCG22 gene. Timepoint 3

```{r}
Toptp3_heu_huu.sex.race.ct5.HCG22 <- Toptp3_heu_huu.sex.race.ct5[grepl("HCG22", Toptp3_heu_huu.sex.race.ct5$UCSC_RefGene_Name),]

table(Toptp3_heu_huu.sex.race.ct5.HCG22[,14] < 0.05/(30))
```

0 sites were found.

```{r}
Toptp3_heu_huu.sex.race.ct5.HCG22.bonf <- Toptp3_heu_huu.sex.race.ct5.HCG22[which(Toptp3_heu_huu.sex.race.ct5.HCG22$P.value < 0.05/(30)),]

cpg_hist <- Toptp3_heu_huu.sex.race.ct5.HCG22.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with HCG22 \n between heu+ and HUU group at timepoint 3")
```


#EBF4 - Timepoint 12 - HIV vs. HEU
Target the EBF4 gene. Timepoint 12


```{r}
Toptp12_hiv_heu.sex.race.ct3.EBF4 <- Toptp12_hiv_heu.sex.race.ct3[grepl("EBF4", Toptp12_hiv_heu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_hiv_heu.sex.race.ct3.EBF4[,14] < 0.05/(56))
```

0 sites were found.

```{r}
Toptp12_hiv_heu.sex.race.ct3.EBF4.bonf <- Toptp12_hiv_heu.sex.race.ct3.EBF4[which(Toptp12_hiv_heu.sex.race.ct3.EBF4$P.value < 0.05/(56)),]

cpg_hist <- Toptp12_hiv_heu.sex.race.ct3.EBF4.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with EBF4 \n between HIV+ and HEU group at timepoint 12")
```

Timepoint 12
HIV vs. HEU
Target the XDH gene. Timepoint 2

```{r}
Toptp12_hiv_heu.sex.race.ct3.XDH <- Toptp12_hiv_heu.sex.race.ct3[grepl("XDH", Toptp12_hiv_heu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_hiv_heu.sex.race.ct3.XDH[,14] < 0.05/(21))
```

1 site was found.

```{r}
Toptp12_hiv_heu.sex.race.ct3.XDH.bonf <- Toptp12_hiv_heu.sex.race.ct3.XDH[which(Toptp12_hiv_heu.sex.race.ct3.XDH$P.value < 0.05/(21)),]

cpg_hist <- Toptp12_hiv_heu.sex.race.ct3.XDH.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with XDH \n between HIV+ and HEU group at timepoint 12")
```

Save results.

```{r}

genename.XDH <- Toptp12_hiv_heu.sex.race.ct3.XDH$CPG.Labels
average.XDH <- data.frame(HIV = rep(0, length(genename.XDH)), HEU = rep(0, length(genename.XDH)))
eawp <- getEAWP(betas.new.tp12_hiv_heu)
for (i in 1:length(genename.XDH)){
  average.XDH[i,] <- tapply(eawp$exprs[genename.XDH[i],], pheno.tp12_hiv_heu$Group, mean, na.rm=TRUE)
}
average.XDH$CPG.Labels <- genename.XDH
toptp12_hiv_heu.sex.race.ct3.XDH <- left_join(Toptp12_hiv_heu.sex.race.ct3.XDH, average.XDH, by="CPG.Labels")

write.csv(toptp12_hiv_heu.sex.race.ct3.XDH, file = "C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\Jasmine\\Aim 1\\toptp12_hiv_heu.sex.race.ct3.XDH.csv")

```


Timepoint 12
HIV vs. HEU
Target the SPERT gene. Timepoint 2


```{r}
Toptp12_hiv_heu.sex.race.ct3.SPERT <- Toptp12_hiv_heu.sex.race.ct3[grepl("SPERT", Toptp12_hiv_heu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_hiv_heu.sex.race.ct3.SPERT[,14] < 0.05/(17))
```

2 sites were found.

```{r}
Toptp12_hiv_heu.sex.race.ct3.SPERT.bonf <- Toptp12_hiv_heu.sex.race.ct3.SPERT[which(Toptp12_hiv_heu.sex.race.ct3.SPERT$P.value < 0.05/(17)),]

cpg_hist <- Toptp12_hiv_heu.sex.race.ct3.SPERT.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with SPERT \n between HIV+ and HEU group at timepoint 12")
```

Save Results.

```{r}
genename.SPERT <- Toptp12_hiv_heu.sex.race.ct3.SPERT$CPG.Labels
average.SPERT <- data.frame(HIV = rep(0, length(genename.SPERT)), HEU = rep(0, length(genename.SPERT)))
eawp <- getEAWP(betas.new.tp12_hiv_heu)
for (i in 1:length(genename.SPERT)){
  average.SPERT[i,] <- tapply(eawp$exprs[genename.SPERT[i],], pheno.tp12_hiv_heu$Group, mean, na.rm=TRUE)
}
average.SPERT$CPG.Labels <- genename.SPERT
toptp12_hiv_heu.sex.race.ct3.SPERT <- left_join(Toptp12_hiv_heu.sex.race.ct3.SPERT, average.SPERT, by="CPG.Labels")

write.csv(toptp12_hiv_heu.sex.race.ct3.SPERT, file = "C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\Jasmine\\Aim 1\\toptp12_hiv_heu.sex.race.ct3.SPERT.csv")
```


Timepoint 12
HIV vs. HEU
Target the NLRC5 gene. Timepoint 2

```{r}
Toptp12_hiv_heu.sex.race.ct3.NLRC5 <- Toptp12_hiv_heu.sex.race.ct3[grepl("NLRC5", Toptp12_hiv_heu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_hiv_heu.sex.race.ct3.NLRC5[,14] < 0.05/(62))

```

4 sites were found.

```{r}
Toptp12_hiv_heu.sex.race.ct3.NLRC5.bonf <- Toptp12_hiv_heu.sex.race.ct3.NLRC5[which(Toptp12_hiv_heu.sex.race.ct3.NLRC5$P.value < 0.05/(62)),]

cpg_hist <- Toptp12_hiv_heu.sex.race.ct3.NLRC5.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with NLRC5 \n between HIV+ and HEU group at timepoint 12")

```


Save Results.

```{r}
genename.NLRC5 <- Toptp12_hiv_heu.sex.race.ct3.NLRC5$CPG.Labels
average.NLRC5 <- data.frame(HIV = rep(0, length(genename.NLRC5)), HEU = rep(0, length(genename.NLRC5)))
eawp <- getEAWP(betas.new.tp12_hiv_heu)
for (i in 1:length(genename.NLRC5)){
  average.NLRC5[i,] <- tapply(eawp$exprs[genename.NLRC5[i],], pheno.tp12_hiv_heu$Group, mean, na.rm=TRUE)
}
average.NLRC5$CPG.Labels <- genename.NLRC5
toptp12_hiv_heu.sex.race.ct3.NLRC5 <- left_join(Toptp12_hiv_heu.sex.race.ct3.NLRC5, average.NLRC5, by="CPG.Labels")

write.csv(toptp12_hiv_heu.sex.race.ct3.NLRC5, file = "C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\Jasmine\\Aim 1\\toptp12_hiv_heu.sex.race.ct3.NLRC5.csv")

```



Timepoint 12
HIV vs. HEU
Target the FOXP1 gene. Timepoint 2

```{r}
Toptp12_hiv_heu.sex.race.ct3.FOXP1 <- Toptp12_hiv_heu.sex.race.ct3[grepl("FOXP1", Toptp12_hiv_heu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_hiv_heu.sex.race.ct3.FOXP1[,14] < 0.05/(299))

```

2 sites were found.

```{r}
Toptp12_hiv_heu.sex.race.ct3.FOXP1.bonf <- Toptp12_hiv_heu.sex.race.ct3.FOXP1[which(Toptp12_hiv_heu.sex.race.ct3.FOXP1$P.value < 0.05/(299)),]

cpg_hist <- Toptp12_hiv_heu.sex.race.ct3.FOXP1.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with FOXP1 \n between HIV+ and HEU group at timepoint 12")

```

Save results.

```{r}
genename.FOXP1 <- Toptp12_hiv_heu.sex.race.ct3.FOXP1$CPG.Labels
average.FOXP1 <- data.frame(HIV = rep(0, length(genename.FOXP1)), HEU = rep(0, length(genename.FOXP1)))
eawp <- getEAWP(betas.new.tp12_hiv_heu)
for (i in 1:length(genename.FOXP1)){
  average.FOXP1[i,] <- tapply(eawp$exprs[genename.FOXP1[i],], pheno.tp12_hiv_heu$Group, mean, na.rm=TRUE)
}
average.FOXP1$CPG.Labels <- genename.FOXP1
toptp12_hiv_heu.sex.race.ct3.FOXP1 <- left_join(Toptp12_hiv_heu.sex.race.ct3.FOXP1, average.FOXP1, by="CPG.Labels")

write.csv(toptp12_hiv_heu.sex.race.ct3.FOXP1, file = "C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\Jasmine\\Aim 1\\toptp12_hiv_heu.sex.race.ct3.FOXP1.csv")
```





Timepoint 12
HIV vs. HEU
Target the FNDC3B gene. Timepoint 2

```{r}
Toptp12_hiv_heu.sex.race.ct3.FNDC3B <- Toptp12_hiv_heu.sex.race.ct3[grepl("FNDC3B", Toptp12_hiv_heu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_hiv_heu.sex.race.ct3.FNDC3B[,14] < 0.05/(152))
```

0 sites were found.
```{r}
Toptp12_hiv_heu.sex.race.ct3.FNDC3B.bonf <- Toptp12_hiv_heu.sex.race.ct3.FNDC3B[which(Toptp12_hiv_heu.sex.race.ct3.FNDC3B$P.value < 0.05/(152)),]

cpg_hist <- Toptp12_hiv_heu.sex.race.ct3.FNDC3B.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with FNDC3B \n between HIV+ and HEU group at timepoint 12")

```


Timepoint 12
HIV vs. HEU
Target the RPS6KA2 gene. Timepoint 2

```{r}
Toptp12_hiv_heu.sex.race.ct3.RPS6KA2 <- Toptp12_hiv_heu.sex.race.ct3[grepl("RPS6KA2", Toptp12_hiv_heu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_hiv_heu.sex.race.ct3.RPS6KA2[,14] < 0.05/(362))
```

2 sites were found.

```{r}
Toptp12_hiv_heu.sex.race.ct3.RPS6KA2.bonf <- Toptp12_hiv_heu.sex.race.ct3.RPS6KA2[which(Toptp12_hiv_heu.sex.race.ct3.RPS6KA2$P.value < 0.05/(362)),]

cpg_hist <- Toptp12_hiv_heu.sex.race.ct3.RPS6KA2.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with RPS6KA2 \n between HIV+ and HEU group at timepoint 12")

```

Save results.
```{r}
genename.RPS6KA2 <- Toptp12_hiv_heu.sex.race.ct3.RPS6KA2$CPG.Labels
average.RPS6KA2 <- data.frame(HIV = rep(0, length(genename.RPS6KA2)), HEU = rep(0, length(genename.RPS6KA2)))
eawp <- getEAWP(betas.new.tp12_hiv_heu)
for (i in 1:length(genename.RPS6KA2)){
  average.RPS6KA2[i,] <- tapply(eawp$exprs[genename.RPS6KA2[i],], pheno.tp12_hiv_heu$Group, mean, na.rm=TRUE)
}
average.RPS6KA2$CPG.Labels <- genename.RPS6KA2
toptp12_hiv_heu.sex.race.ct3.RPS6KA2 <- left_join(Toptp12_hiv_heu.sex.race.ct3.RPS6KA2, average.RPS6KA2, by="CPG.Labels")

write.csv(toptp12_hiv_heu.sex.race.ct3.RPS6KA2, file = "C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\Jasmine\\Aim 1\\toptp12_hiv_heu.sex.race.ct3.RPS6KA2.csv")
```


Timepoint 12
HIV vs. HEU
Target the DLL1 gene. Timepoint 2

```{r}
Toptp12_hiv_heu.sex.race.ct3.DLL1 <- Toptp12_hiv_heu.sex.race.ct3[grepl("DLL1", Toptp12_hiv_heu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_hiv_heu.sex.race.ct3.DLL1[,14] < 0.05/(38))
```

0 sites were found.
```{r}
Toptp12_hiv_heu.sex.race.ct3.DLL1.bonf <- Toptp12_hiv_heu.sex.race.ct3.DLL1[which(Toptp12_hiv_heu.sex.race.ct3.DLL1$P.value < 0.05/(38)),]

cpg_hist <- Toptp12_hiv_heu.sex.race.ct3.DLL1.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with DLL1 \n between HIV+ and HEU group at timepoint 12")
```

Timepoint 12
HIV vs. HEU
Target the HCG22 gene. Timepoint 2

```{r}
Toptp12_hiv_heu.sex.race.ct3.HCG22 <- Toptp12_hiv_heu.sex.race.ct3[grepl("HCG22", Toptp12_hiv_heu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_hiv_heu.sex.race.ct3.HCG22[,14] < 0.05/(30))
```

1 site was found.

```{r}
Toptp12_hiv_heu.sex.race.ct3.HCG22.bonf <- Toptp12_hiv_heu.sex.race.ct3.HCG22[which(Toptp12_hiv_heu.sex.race.ct3.HCG22$P.value < 0.05/(30)),]

cpg_hist <- Toptp12_hiv_heu.sex.race.ct3.HCG22.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with HCG22 \n between HIV+ and HEU group at timepoint 12")
```

```{r}
genename.HCG22 <- Toptp12_hiv_heu.sex.race.ct3.HCG22$CPG.Labels
average.HCG22 <- data.frame(HIV = rep(0, length(genename.HCG22)), HEU = rep(0, length(genename.HCG22)))
eawp <- getEAWP(betas.new.tp12_hiv_heu)
for (i in 1:length(genename.HCG22)){
  average.HCG22[i,] <- tapply(eawp$exprs[genename.HCG22[i],], pheno.tp12_hiv_heu$Group, mean, na.rm=TRUE)
}
average.HCG22$CPG.Labels <- genename.HCG22
toptp12_hiv_heu.sex.race.ct3.HCG22 <- left_join(Toptp12_hiv_heu.sex.race.ct3.HCG22, average.HCG22, by="CPG.Labels")

write.csv(toptp12_hiv_heu.sex.race.ct3.HCG22, file = "C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\Jasmine\\Aim 1\\toptp12_hiv_heu.sex.race.ct3.HCG22.csv")
```


#EFB4 - Timepoint 12 - HIV vs. HUU
Target the EBF4 gene. Timepoint 2.

```{r}
Toptp12_hiv_huu.sex.race.ct3.EBF4 <- Toptp12_hiv_huu.sex.race.ct3[grepl("EBF4", Toptp12_hiv_huu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_hiv_huu.sex.race.ct3.EBF4[,14] < 0.05/(56))
```


0 sites were found.

```{r}
Toptp12_hiv_huu.sex.race.ct3.EBF4.bonf <- Toptp12_hiv_huu.sex.race.ct3.EBF4[which(Toptp12_hiv_huu.sex.race.ct3.EBF4$P.value < 0.05/(56)),]

cpg_hist <- Toptp12_hiv_huu.sex.race.ct3.EBF4.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with EBF4 \n between HIV+ and HUU group at timepoint 12")

```


Timepoint 12
HIV vs. HUU
Target the XDH gene. Timepoint 2

```{r}
Toptp12_hiv_huu.sex.race.ct3.XDH <- Toptp12_hiv_huu.sex.race.ct3[grepl("XDH", Toptp12_hiv_huu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_hiv_huu.sex.race.ct3.XDH[,14] < 0.05/(21))
```

1 site was found.
```{r}
Toptp12_hiv_huu.sex.race.ct3.XDH.bonf <- Toptp12_hiv_huu.sex.race.ct3.XDH[which(Toptp12_hiv_huu.sex.race.ct3.XDH$P.value < 0.05/(21)),]

cpg_hist <- Toptp12_hiv_huu.sex.race.ct3.XDH.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with XDH \n between HIV+ and HUU group at timepoint 12")
```

Save results.

```{r}
genename.XDH <- Toptp12_hiv_huu.sex.race.ct3.XDH$CPG.Labels
average.XDH <- data.frame(HIV = rep(0, length(genename.XDH)), HUU = rep(0, length(genename.XDH)))
eawp <- getEAWP(betas.new.tp12_hiv_huu)
for (i in 1:length(genename.XDH)){
  average.XDH[i,] <- tapply(eawp$exprs[genename.XDH[i],], pheno.tp12_hiv_huu$Group, mean, na.rm=TRUE)
}
average.XDH$CPG.Labels <- genename.XDH
toptp12_hiv_huu.sex.race.ct3.XDH <- left_join(Toptp12_hiv_huu.sex.race.ct3.XDH, average.XDH, by="CPG.Labels")

write.csv(toptp12_hiv_huu.sex.race.ct3.XDH, file = "C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\Jasmine\\Aim 1\\toptp12_hiv_huu.sex.race.ct3.XDH.csv")
```



Timepoint 12
HIV vs. HUU
Target the SPERT gene. Timepoint 2

```{r}
Toptp12_hiv_huu.sex.race.ct3.SPERT <- Toptp12_hiv_huu.sex.race.ct3[grepl("SPERT", Toptp12_hiv_huu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_hiv_huu.sex.race.ct3.SPERT[,14] < 0.05/(17))
```

2 sites were found.

```{r}
Toptp12_hiv_huu.sex.race.ct3.SPERT.bonf <- Toptp12_hiv_huu.sex.race.ct3.SPERT[which(Toptp12_hiv_huu.sex.race.ct3.SPERT$P.value < 0.05/(17)),]

cpg_hist <- Toptp12_hiv_huu.sex.race.ct3.SPERT.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with SPERT \n between HIV+ and HUU group at timepoint 12")

```

```{r}
genename.SPERT <- Toptp12_hiv_huu.sex.race.ct3.SPERT$CPG.Labels
average.SPERT <- data.frame(HIV = rep(0, length(genename.SPERT)), HUU = rep(0, length(genename.SPERT)))
eawp <- getEAWP(betas.new.tp12_hiv_huu)
for (i in 1:length(genename.SPERT)){
  average.SPERT[i,] <- tapply(eawp$exprs[genename.SPERT[i],], pheno.tp12_hiv_huu$Group, mean, na.rm=TRUE)
}
average.SPERT$CPG.Labels <- genename.SPERT
toptp12_hiv_huu.sex.race.ct3.SPERT <- left_join(Toptp12_hiv_huu.sex.race.ct3.SPERT, average.SPERT, by="CPG.Labels")

write.csv(toptp12_hiv_huu.sex.race.ct3.SPERT, file = "C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\Jasmine\\Aim 1\\toptp12_hiv_huu.sex.race.ct3.SPERT.csv")

```




Timepoint 12
HIV vs. HUU
Target the NLRC5 gene. Timepoint 2


```{r}
Toptp12_hiv_huu.sex.race.ct3.NLRC5 <- Toptp12_hiv_huu.sex.race.ct3[grepl("NLRC5", Toptp12_hiv_huu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_hiv_huu.sex.race.ct3.NLRC5[,14] < 0.05/(62))
```

2 sites were found. 

```{r}
Toptp12_hiv_huu.sex.race.ct3.NLRC5.bonf <- Toptp12_hiv_huu.sex.race.ct3.NLRC5[which(Toptp12_hiv_huu.sex.race.ct3.NLRC5$P.value < 0.05/(62)),]

cpg_hist <- Toptp12_hiv_huu.sex.race.ct3.NLRC5.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with NLRC5 \n between HIV+ and HUU group at timepoint 12")

```

Save results.

```{r}
genename.NLRC5 <- Toptp12_hiv_huu.sex.race.ct3.NLRC5$CPG.Labels
average.NLRC5 <- data.frame(HIV = rep(0, length(genename.NLRC5)), HUU = rep(0, length(genename.NLRC5)))
eawp <- getEAWP(betas.new.tp12_hiv_huu)
for (i in 1:length(genename.NLRC5)){
  average.NLRC5[i,] <- tapply(eawp$exprs[genename.NLRC5[i],], pheno.tp12_hiv_huu$Group, mean, na.rm=TRUE)
}
average.NLRC5$CPG.Labels <- genename.NLRC5
toptp12_hiv_huu.sex.race.ct3.NLRC5 <- left_join(Toptp12_hiv_huu.sex.race.ct3.NLRC5, average.NLRC5, by="CPG.Labels")

write.csv(toptp12_hiv_huu.sex.race.ct3.NLRC5, file = "C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\Jasmine\\Aim 1\\toptp12_hiv_huu.sex.race.ct3.NLRC5.csv")

```



Timepoint 12
HIV vs. HUU
Target the FOXP1 gene. Timepoint 2


```{r}
Toptp12_hiv_huu.sex.race.ct3.FOXP1 <- Toptp12_hiv_huu.sex.race.ct3[grepl("FOXP1", Toptp12_hiv_huu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_hiv_huu.sex.race.ct3.FOXP1[,14] < 0.05/(299))
```

1 site was found.

```{r}
Toptp12_hiv_huu.sex.race.ct3.FOXP1.bonf <- Toptp12_hiv_huu.sex.race.ct3.FOXP1[which(Toptp12_hiv_huu.sex.race.ct3.FOXP1$P.value < 0.05/(299)),]

cpg_hist <- Toptp12_hiv_huu.sex.race.ct3.FOXP1.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with FOXP1 \n between HIV+ and HUU group at timepoint 12")
```


```{r}
genename.FOXP1 <- Toptp12_hiv_huu.sex.race.ct3.FOXP1$CPG.Labels
average.FOXP1 <- data.frame(HIV = rep(0, length(genename.FOXP1)), HUU = rep(0, length(genename.FOXP1)))
eawp <- getEAWP(betas.new.tp12_hiv_huu)
for (i in 1:length(genename.FOXP1)){
  average.FOXP1[i,] <- tapply(eawp$exprs[genename.FOXP1[i],], pheno.tp12_hiv_huu$Group, mean, na.rm=TRUE)
}
average.FOXP1$CPG.Labels <- genename.FOXP1
toptp12_hiv_huu.sex.race.ct3.FOXP1 <- left_join(Toptp12_hiv_huu.sex.race.ct3.FOXP1, average.FOXP1, by="CPG.Labels")

write.csv(toptp12_hiv_huu.sex.race.ct3.FOXP1, file = "C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\Jasmine\\Aim 1\\toptp12_hiv_huu.sex.race.ct3.FOXP1.csv")
```



Timepoint 12
HIV vs. HUU
Target the FNDC3B gene. Timepoint 2

```{r}
Toptp12_hiv_huu.sex.race.ct3.FNDC3B <- Toptp12_hiv_huu.sex.race.ct3[grepl("FNDC3B", Toptp12_hiv_huu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_hiv_huu.sex.race.ct3.FNDC3B[,14] < 0.05/(152))
```
0 sites were found.

```{r}
Toptp12_hiv_huu.sex.race.ct3.FNDC3B.bonf <- Toptp12_hiv_huu.sex.race.ct3.FNDC3B[which(Toptp12_hiv_huu.sex.race.ct3.FNDC3B$P.value < 0.05/(152)),]

cpg_hist <- Toptp12_hiv_huu.sex.race.ct3.FNDC3B.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with FNDC3B \n between HIV+ and HUU group at timepoint 12")
```


Timepoint 12
HIV vs. HUU
Target the RPS6KA2 gene. Timepoint 2

```{r}
Toptp12_hiv_huu.sex.race.ct3.RPS6KA2 <- Toptp12_hiv_huu.sex.race.ct3[grepl("RPS6KA2", Toptp12_hiv_huu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_hiv_huu.sex.race.ct3.RPS6KA2[,14] < 0.05/(362))
```

0 sites were found.

```{r}
Toptp12_hiv_huu.sex.race.ct3.RPS6KA2.bonf <- Toptp12_hiv_huu.sex.race.ct3.RPS6KA2[which(Toptp12_hiv_huu.sex.race.ct3.RPS6KA2$P.value < 0.05/(362)),]

cpg_hist <- Toptp12_hiv_huu.sex.race.ct3.RPS6KA2.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with RPS6KA2 \n between HIV+ and HUU group at timepoint 12")

```


Timepoint 12
HIV vs. HUU
Target the DLL1 gene. Timepoint 2

```{r}
Toptp12_hiv_huu.sex.race.ct3.DLL1 <- Toptp12_hiv_huu.sex.race.ct3[grepl("DLL1", Toptp12_hiv_huu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_hiv_huu.sex.race.ct3.DLL1[,14] < 0.05/(38))
```

0 sites were found.

```{r}
Toptp12_hiv_huu.sex.race.ct3.DLL1.bonf <- Toptp12_hiv_huu.sex.race.ct3.DLL1[which(Toptp12_hiv_huu.sex.race.ct3.DLL1$P.value < 0.05/(38)),]

cpg_hist <- Toptp12_hiv_huu.sex.race.ct3.DLL1.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with DLL1 \n between HIV+ and HUU group at timepoint 12")
```


Timepoint 12
HIV vs. HUU
Target the HCG22 gene. Timepoint 2

```{r}
Toptp12_hiv_huu.sex.race.ct3.HCG22 <- Toptp12_hiv_huu.sex.race.ct3[grepl("HCG22", Toptp12_hiv_huu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_hiv_huu.sex.race.ct3.HCG22[,14] < 0.05/(30))
```

1 site was found.

```{r}
Toptp12_hiv_huu.sex.race.ct3.HCG22.bonf <- Toptp12_hiv_huu.sex.race.ct3.HCG22[which(Toptp12_hiv_huu.sex.race.ct3.HCG22$P.value < 0.05/(30)),]

cpg_hist <- Toptp12_hiv_huu.sex.race.ct3.HCG22.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with HCG22 \n between HIV+ and HUU group at timepoint 12")

```

```{r}
genename.HCG22 <- Toptp12_hiv_huu.sex.race.ct3.HCG22$CPG.Labels
average.HCG22 <- data.frame(HIV = rep(0, length(genename.HCG22)), HUU = rep(0, length(genename.HCG22)))
eawp <- getEAWP(betas.new.tp12_hiv_huu)
for (i in 1:length(genename.HCG22)){
  average.HCG22[i,] <- tapply(eawp$exprs[genename.HCG22[i],], pheno.tp12_hiv_huu$Group, mean, na.rm=TRUE)
}
average.HCG22$CPG.Labels <- genename.HCG22
toptp12_hiv_huu.sex.race.ct3.HCG22 <- left_join(Toptp12_hiv_huu.sex.race.ct3.HCG22, average.HCG22, by="CPG.Labels")

write.csv(toptp12_hiv_huu.sex.race.ct3.HCG22, file = "C:\\Users\\jasmi\\Box\\RU MICS (Shared)\\Jasmine\\Aim 1\\toptp12_hiv_huu.sex.race.ct3.HCG22.csv")

```



# EFB4 - Timepoint 12 - HEU vs. HUU

Target the EBF4 gene. Timepoint 2.

```{r}
Toptp12_heu_huu.sex.race.ct3.EBF4 <- Toptp12_heu_huu.sex.race.ct3[grepl("EBF4", Toptp12_heu_huu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_heu_huu.sex.race.ct3.EBF4[,14] < 0.05/(56))
```


0 sites were found.

```{r}
Toptp12_heu_huu.sex.race.ct3.EBF4.bonf <- Toptp12_heu_huu.sex.race.ct3.EBF4[which(Toptp12_heu_huu.sex.race.ct3.EBF4$P.value < 0.05/(56)),]

cpg_hist <- Toptp12_heu_huu.sex.race.ct3.EBF4.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with EBF4 \n between heu+ and HUU group at timepoint 12")

```


Timepoint 12

heu vs. HUU
Target the XDH gene. Timepoint 2

```{r}
Toptp12_heu_huu.sex.race.ct3.XDH <- Toptp12_heu_huu.sex.race.ct3[grepl("XDH", Toptp12_heu_huu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_heu_huu.sex.race.ct3.XDH[,14] < 0.05/(21))
```

0 sites were found.

```{r}
Toptp12_heu_huu.sex.race.ct3.XDH.bonf <- Toptp12_heu_huu.sex.race.ct3.XDH[which(Toptp12_heu_huu.sex.race.ct3.XDH$P.value < 0.05/(21)),]

cpg_hist <- Toptp12_heu_huu.sex.race.ct3.XDH.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with XDH \n between heu+ and HUU group at timepoint 12")
```


Timepoint 12

heu vs. HUU
Target the SPERT gene. Timepoint 2

```{r}
Toptp12_heu_huu.sex.race.ct3.SPERT <- Toptp12_heu_huu.sex.race.ct3[grepl("SPERT", Toptp12_heu_huu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_heu_huu.sex.race.ct3.SPERT[,14] < 0.05/(17))
```

0 sites were found.

```{r}
Toptp12_heu_huu.sex.race.ct3.SPERT.bonf <- Toptp12_heu_huu.sex.race.ct3.SPERT[which(Toptp12_heu_huu.sex.race.ct3.SPERT$P.value < 0.05/(17)),]

cpg_hist <- Toptp12_heu_huu.sex.race.ct3.SPERT.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with SPERT \n between heu+ and HUU group at timepoint 12")

```


Timepoint 12
heu vs. HUU
Target the NLRC5 gene. Timepoint 2


```{r}
Toptp12_heu_huu.sex.race.ct3.NLRC5 <- Toptp12_heu_huu.sex.race.ct3[grepl("NLRC5", Toptp12_heu_huu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_heu_huu.sex.race.ct3.NLRC5[,14] < 0.05/(62))
```

0 sites were found.

```{r}
Toptp12_heu_huu.sex.race.ct3.NLRC5.bonf <- Toptp12_heu_huu.sex.race.ct3.NLRC5[which(Toptp12_heu_huu.sex.race.ct3.NLRC5$P.value < 0.05/(62)),]

cpg_hist <- Toptp12_heu_huu.sex.race.ct3.NLRC5.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with NLRC5 \n between heu+ and HUU group at timepoint 12")

```


Timepoint 12
heu vs. HUU
Target the FOXP1 gene. Timepoint 2


```{r}
Toptp12_heu_huu.sex.race.ct3.FOXP1 <- Toptp12_heu_huu.sex.race.ct3[grepl("FOXP1", Toptp12_heu_huu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_heu_huu.sex.race.ct3.FOXP1[,14] < 0.05/(299))
```

0 sites were found.

```{r}
Toptp12_heu_huu.sex.race.ct3.FOXP1.bonf <- Toptp12_heu_huu.sex.race.ct3.FOXP1[which(Toptp12_heu_huu.sex.race.ct3.FOXP1$P.value < 0.05/(299)),]

cpg_hist <- Toptp12_heu_huu.sex.race.ct3.FOXP1.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with FOXP1 \n between heu+ and HUU group at timepoint 12")
```


Timepoint 12
heu vs. HUU
Target the FNDC3B gene. Timepoint 2

```{r}
Toptp12_heu_huu.sex.race.ct3.FNDC3B <- Toptp12_heu_huu.sex.race.ct3[grepl("FNDC3B", Toptp12_heu_huu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_heu_huu.sex.race.ct3.FNDC3B[,14] < 0.05/(152))
```

0 sites were found.

```{r}
Toptp12_heu_huu.sex.race.ct3.FNDC3B.bonf <- Toptp12_heu_huu.sex.race.ct3.FNDC3B[which(Toptp12_heu_huu.sex.race.ct3.FNDC3B$P.value < 0.05/(152)),]

cpg_hist <- Toptp12_heu_huu.sex.race.ct3.FNDC3B.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with FNDC3B \n between heu+ and HUU group at timepoint 12")
```


Timepoint 12
heu vs. HUU
Target the RPS6KA2 gene. Timepoint 2

```{r}
Toptp12_heu_huu.sex.race.ct3.RPS6KA2 <- Toptp12_heu_huu.sex.race.ct3[grepl("RPS6KA2", Toptp12_heu_huu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_heu_huu.sex.race.ct3.RPS6KA2[,14] < 0.05/(362))
```

0 sites were found.

```{r}
Toptp12_heu_huu.sex.race.ct3.RPS6KA2.bonf <- Toptp12_heu_huu.sex.race.ct3.RPS6KA2[which(Toptp12_heu_huu.sex.race.ct3.RPS6KA2$P.value < 0.05/(362)),]

cpg_hist <- Toptp12_heu_huu.sex.race.ct3.RPS6KA2.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with RPS6KA2 \n between heu+ and HUU group at timepoint 12")

```


Timepoint 12
heu vs. HUU
Target the DLL1 gene. Timepoint 2

```{r}
Toptp12_heu_huu.sex.race.ct3.DLL1 <- Toptp12_heu_huu.sex.race.ct3[grepl("DLL1", Toptp12_heu_huu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_heu_huu.sex.race.ct3.DLL1[,14] < 0.05/(38))
```

0 sites were found.

```{r}
Toptp12_heu_huu.sex.race.ct3.DLL1.bonf <- Toptp12_heu_huu.sex.race.ct3.DLL1[which(Toptp12_heu_huu.sex.race.ct3.DLL1$P.value < 0.05/(38)),]

cpg_hist <- Toptp12_heu_huu.sex.race.ct3.DLL1.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with DLL1 \n between heu+ and HUU group at timepoint 12")
```


Timepoint 12
heu vs. HUU
Target the HCG22 gene. Timepoint 2

```{r}
Toptp12_heu_huu.sex.race.ct3.HCG22 <- Toptp12_heu_huu.sex.race.ct3[grepl("HCG22", Toptp12_heu_huu.sex.race.ct3$UCSC_RefGene_Name),]

table(Toptp12_heu_huu.sex.race.ct3.HCG22[,14] < 0.05/(30))
```

0 sites were found.

```{r}
Toptp12_heu_huu.sex.race.ct3.HCG22.bonf <- Toptp12_heu_huu.sex.race.ct3.HCG22[which(Toptp12_heu_huu.sex.race.ct3.HCG22$P.value < 0.05/(30)),]

cpg_hist <- Toptp12_heu_huu.sex.race.ct3.HCG22.bonf[,1:2]
cpg_hist$CPG.Labels <- factor(cpg_hist$CPG.Labels,levels = rev(cpg_hist$CPG.Labels))
ggplot(data = cpg_hist, aes(x = CPG.Labels , y = logFC)) + 
  geom_bar(aes(fill = logFC>0), stat = "identity", show.legend = F) + coord_flip() +
  ggtitle("logFC of the CpG sites associated with HCG22 \n between heu+ and HUU group at timepoint 12")


```



















